---
title: "103-global-analysis"
output: html_document
date: "2024-04-02"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(poppr)
library(vcfR)
library(phytools)
library(phangorn)
library(ape)
library(ggtree)
library(tanggle)
```

We can try to generate a phylogenetic network of all our samples.   

We need to be identify key lineages.    

We also need to account for heterogeneity in coverage by downsampling to a certain threshold.   


Our key lineages are Eremichthys, Siphateles.

May not have euchila, from Hubbs and Miller (1972), Fish Creek Spring, Fish Creek Valley, White Pine Co., NV (synonym of newarkensis)
species:   
mohavensis (no subs)
isolatata (no subs)
newarkensis (newarkensis, euchila)
thalassinus (thalassinus, vaccaceps)
bicolor (bicolor, columbianus, eurysoma, 'Silver Lake Valley')
obesus (obesus, snyderi, oregonensis, pectinifer)
'toikona'?

```{r}
m<-read_csv("meta/04032023-edited.csv") %>% mutate(Path=paste0("data/align/",Ind,".sort.flt.bam"))
m
```

```{r}
m5<-m %>% filter(Dedup>5e5)
m5 %>% select(Path) %>% write_tsv("bamlists/test1124.bamlist", col_names = FALSE)
m5 %>% group_by(Species) %>% summarize(Count=n())
```

We should be able to sort these rapidly into
(1) Eremichthys
(2) Siphateles 

1124 samples.  Using angsd with 24 processors and 128 G mem like so:

```{sh, eval=FALSE}
srun -p bigmemh -t 36:00:00 --mem=128G --nodes=1 --ntasks=1 --cpus-per-task=24  $HOME/angsd/angsd -P 24  \
-bam bamlists/test1124.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-minInd 1012 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/103/test1124-ibs-90 >outputs/103/test1124-ibs-90.out 2> outputs/103/test1124-ibs-90.err &

srun -p bigmemh -t 36:00:00 --mem=128G --nodes=1 --ntasks=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/test1124.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-minInd 1012 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/103/test1124-12cpu-ibs-90 >outputs/103/test1124-12cpu-ibs-90.out 2> outputs/103/test1124-12cpu-ibs-90.err &

```

12 CPU one the race! Only by a couple of minutes.

Have 114,331 GLs.


```{r}
m <- as.matrix(read.table("outputs/103/test1124-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(m5)
#text1

ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Species), pch=21, cex=3, alpha=0.75) +
  #geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("Genome-Wide PCA of 1124 Samples") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) 
```
```{r}
ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Species), pch=21, cex=3, alpha=0.75) +
  #geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("Genome-Wide PCA of 1124 Samples") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))  +
  facet_wrap(.~Species)

ggsave("outputs/103/pca-1124.pdf")
```
Mostly seems to sort out.
```{r}
ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Species), pch=21, cex=3, alpha=0.75) +
  #geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("Genome-Wide PCA of 1124 Samples") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) 
```
```{r}
ones<-covs %>% filter(Species=="obesus") %>% filter(V1 < 0.010)
twos<-covs %>% filter(Species=="obesus") %>% filter(V2 > 0.005)
ones
twos
```

Alkali liake stuff is weird. 

```{r}
ggplot(covs %>% filter(Species %in% c("obesus")) %>% filter(!SampleName %in% c(ones$SampleName, twos$SampleName))) +
  geom_point(aes(x=V1, y=V2, fill=Vicinity), pch=21, cex=3, alpha=0.75) +
  #geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("Genome-Wide PCA of 1124 Samples") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) 
```

May have to split into regions and separate jobs here for SNP calling.    
```{r}
obesus<-m5 %>% filter(Species %in% c("obesus","toikona","snyderi")) %>% filter(!SampleName %in% c(ones$SampleName, twos$SampleName))
write_csv(obesus, file="meta/obesus-500k-meta.csv")
```

```{r}
m<-read_csv("meta/04032023-edited.csv") %>% mutate(Path=paste0("data/align/",Ind,".sort.flt.bam"))
```
```{r}
m %>% filter(Dedup > 1e6) %>% group_by(Species) %>% summarize(Count=n())
```

Downsample.

```{r}
ggplot(m) +
  geom_histogram(aes(x=Dedup)) +
  theme_bw() +
  ylab("Count\n") +
  xlab("\nPost PCR Duplication") +
  theme(panel.grid=element_blank())

```

```{r}
mean(m$Dedup)
median(m$Dedup)
```

```{r}
m %>% filter(Dedup>1e6) %>% summarize(Mean=mean(Dedup), Median=median(Dedup))
```
Downsample to 2e6 reads, 543 ind downsampled.



What I think may work is to:

(1) use >1e6 inds for SNP calls, define major groups
(2) use >5e5 inds for GLs, for pop gen analyses (e.g. with doIbs)



```{r}
down<-m %>%  mutate(Frac=2e6/Dedup)  %>% 
  mutate(Path2=ifelse(Dedup > 2e6, paste0("data/downsample/",Ind,".reduced.bam"),
                     Path))

downsample<-down %>% filter(Dedup > 2e6 ) %>%
  mutate(ReductionCommand = paste0("samtools view -bs ",Frac, " ", Path," > ",
                                   Path2)) 

write_csv(downsample$ReductionCommand %>% as_tibble(), "103.1-downsample.sh", col_names = FALSE)

write_csv(down, "meta/downsampled-paths-meta.csv")
```

544
Downsampling    
```{sh, eval=FALSE}
module load parallel
srun -p high -t 04:00:00 --nodes=1 parallel -j 10 < 103.1-downsample.sh > outputs/103/downsample.stdout 2> outputs/103/downsample.stderr
```

```{r}
million<-down %>% filter(Dedup>1e6) %>% mutate(Sample=paste0(1:n(),"_",1))
million %>% select(Path2) %>% write_tsv("bamlists/test921.bamlist", col_names = FALSE)
million %>% select(GVL_Code) %>% write_tsv("bamlists/test921.names", col_names = FALSE)
million %>% group_by(Species) %>% summarize(Count=n(), MeanReadCounts=mean(Dedup))
```


Ok, now we can do some snp calls and see what shakes out.   
95% threshold.       
```{sh, eval=FALSE}
srun -t 36:00:00 -p bigmemh --mem=128G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 875 -bam bamlists/test921.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-out outputs/103/snps-921  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/103/snps-921.out 2> outputs/103/snps-921.err &

#Reducing MAF to .01
srun -t 36:00:00 -p bigmemh --mem=128G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 875 -bam bamlists/test921.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-out outputs/103/snps-921-01  \
-minMaf 0.01 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/103/snps-921-01.out 2> outputs/103/snps-921-01.err &

```

Creating a vcf
	-> Number of sites retained after filtering: 38682 

```{sh, eval=FALSE}
plink --tped snps-921.tped --tfam snps-921.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz

#renaming vcf with GVL_Code
bcftools reheader --samples bamlists/test921.names -o outputs/103/renamed.vcf.gz outputs/103/plink.vcf.gz
#Need to get +prune module on farm

bcftools +prune -l 0.20 -w 10000 outputs/103/renamed.vcf.gz > outputs/103/pruned.vcf
bcftools +prune -l 0.30 -w 10000 outputs/103/renamed.vcf.gz > outputs/103/pruned-03.vcf


plink --tped snps-921-01.tped --tfam snps-921-01.tfam  --out plink-binary-01 --recode --allow-extra-chr --noweb
plink --ped plink-binary-01.ped --map plink-binary-01.map --recode vcf --allow-extra-chr -out plink-01
bgzip plink-01.vcf 
tabix plink-01.vcf.gz

bcftools reheader --samples bamlists/test921.names -o outputs/103/renamed-01.vcf.gz outputs/103/plink-01.vcf.gz
bcftools +prune -l 0.20 -w 10000 outputs/103/plink-01.vcf.gz > outputs/103/pruned-01.vcf

```

172988 variants and 921 people pass filters and QC @ MAF 0.01
Newark chokes up the thing, so can rename then select one population of Newark fish.

With a phylogeny, can place locations and verify 'species' groupings.    
Then we can try making a nexus file and a NJ tree/network, then a ML tree search one. 

bcftools chokes on renamed.vcf.gz file

```{r}
million %>% select(Sample) %>% write_tsv(file = "bamlists/test921.samples",
                                            col_names = FALSE)

```

```{r}
million %>% filter(Species=="newarkensis") %>% group_by(`Tributary/Collection Location`) %>% summarize(Count=n())
million %>% filter(Species=="mohavensis") %>% group_by(`Tributary/Collection Location`) %>% summarize(Count=n())
nn4<-million %>% filter(Species=="newarkensis") %>% filter(`Tributary/Collection Location`=="NN4")
morning<-million %>% filter(Species=="mohavensis") %>% filter(`Tributary/Collection Location`=="Morningstar Mine")

reducedNew<-million %>% filter(Species != "newarkensis") %>% filter(Species != "mohavensis")
reducedNew<-bind_rows(reducedNew,nn4,morning)
reducedNew %>% select(Sample) %>% write_tsv(file = "bamlists/test921-reducedNew.samples",
                                            col_names = FALSE)
```

```{r}
## Now, to filter out small population sizes
min5<-million %>% group_by(`Tributary/Collection Location`) %>% mutate(Count=n()) %>% filter(Count >= 5) %>% sample_n(weight = Dedup, size = 5)
min5 %>% ungroup() %>% select(Sample)  %>% write_tsv(file = "bamlists/test921-min5.samples",
                                            col_names = FALSE)

#adding more isolatus
iso<-million %>% filter(Species=="isolatus")  %>% sample_n(weight = Dedup, size = 10)
mins<-million %>% filter(Species !="isolatus") %>% filter(`Tributary/Collection Location` != "3/8 Mile Spring") %>% group_by(`Tributary/Collection Location`) %>% mutate(Count=n()) %>% filter(Count >= 5) %>% sample_n(weight = Dedup, size = 5)

min2<-million  %>% filter(`Tributary/Collection Location` != "3/8 Mile Spring") %>% group_by(`Tributary/Collection Location`) %>% mutate(Count=n()) %>% filter(Count >= 2) %>% filter(Count < 5)

min2<-bind_rows(min2, mins, iso)
min2 %>% ungroup() %>% select(Sample)  %>% write_tsv(file = "bamlists/test921-min2.samples",
                                            col_names = FALSE)

```

```{sh, eval=FALSE}
#on my mac
bcftools +prune -l 0.20 -w 10000 outputs/103/plink.vcf.gz > outputs/103/pruned.vcf
~/github/mccloud-rrt/vcf2phylip.py -i pruned.vcf 
seqConverter.pl -dpruned.min4.phy -on

# on u mac 
#local version
bcftools  view -S bamlists/test921.samples outputs/103/plink-01.vcf.gz | bcftools +fill-tags | bcftools view -q 0.01:minor | bcftools +prune -m 0.20 -w 10000 > outputs/103/pruned-01.vcf


#local version
bcftools  view -S bamlists/test921-reducedNew.samples outputs/103/plink-01.vcf.gz | bcftools +fill-tags | bcftools view -q 0.01:minor | bcftools +prune -m 0.20 -w 10000 > outputs/103/pruned-reducedNew-01.vcf

#min5 version

bcftools  view -S bamlists/test921-min5.samples outputs/103/plink-01.vcf.gz | bcftools +fill-tags | bcftools view -q 0.01:minor | bcftools +prune -m 0.20 -w 10000 > outputs/103/pruned-reducedNew-min5-01.vcf

#min2 version

bcftools  view -S bamlists/test921-min2.samples outputs/103/plink-01.vcf.gz | bcftools +fill-tags | bcftools view -q 0.01:minor | bcftools +prune -m 0.20 -w 10000 > outputs/103/pruned-reducedNew-min2-01.vcf


NDA_SUBDIR=osx-64 conda create -n py2 python=2.7 
conda activate py2
~/github/mccloud-rrt/vcf2phylip.py -i pruned-01.vcf 
~/github/mccloud-rrt/vcf2phylip.py -i pruned-reducedNew-01.vcf 
~/github/mccloud-rrt/vcf2phylip.py -i pruned-reducedNew-min5-01.vcf
~/github/mccloud-rrt/vcf2phylip.py -i pruned-reducedNew-min2-01.vcf

```

921 16048 in pruned-01.min4.py.varsites.phy
514 8385 in pruned-reducedNew-01.min4.phy
165 15506 in pruned-reducedNew-min5-01.min4.phy
299 22441  in pruned-reducedNew-min2-01.min4.phy
```{r}

dat<-read.dna(file="outputs/103/pruned-01.min4.phy.varsites.phy")
write.nexus.data(dat, file="outputs/103/pruned-01.nex")

dat<-read.dna(file="outputs/103/pruned-reducedNew-01.min4.phy")
write.nexus.data(dat, file="outputs/103/pruned-reducedNew-01.nex")

dat<-read.dna(file="outputs/103/pruned-reducedNew-min5-01.min4.phy")
write.nexus.data(dat, file="outputs/103/pruned-reducedNew-min5-01.nex")

dat<-read.dna(file="outputs/103/pruned-reducedNew-min2-01.min4.phy")
write.nexus.data(dat, file="outputs/103/pruned-reducedNew-min2-01.nex")

```

Let's make a network in SplitsTree

```{r}
#net<-read.nexus.networx("outputs/103/pruned-reducedNew-01.networx")
#net<-read.nexus.networx("outputs/103/pruned-min5-01.network")
net<-read.nexus.networx("outputs/103/pruned-min2-01.network")
net<-read.nexus.networx("outputs/103/pruned-01.network")
```

```{r}
g<-ggsplitnet(net)  
g$data<-left_join(g$data, million, by=c("label"="Sample")) 

n<-g + 
  geom_tippoint(aes(fill=Species), cex=5, pch=21, alpha=1.0) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22)))

n

ggsave("outputs/103/network.pdf", width=12, height=10)
```
Odd ones include 617_1 634_1
Odd one is DDP13_13 mohave discovery pond
```{r}
million %>% filter(Sample %in% c("617_1","634_1"))
```
(base) maccamp@farm:~/fish-lake/data/raw/split$ cat plate-10_RA_GGAACGTGATTGCAGG.fastq | head -n 400 | grep "^@" -A 1 | perl -pe 's/--//g' | perl -pe 's/@/\>/' > ~/temp.fas
 cat plate-10_RA_GGAACGTGATTGCAGG.fastq | head -n 40000 | grep "^@" -A 1 | perl -pe 's/--//g' | perl -pe 's/@/\>/' > ~/temp.fas

Sample has >= 99.79% alignment success, which is suspicious 
Using pebble scout.... Maybe can find mtDNA in reference genome? Not in reference genome (megablast of sanger-derived mtDNA doesn't work)

ahhh, so no. 

(base) maccamp@farm:~/fish-lake/data/align$ samtools view -b -f 4 plate-10-GGAACGTGATTGCAGG.sort.bam | samtools fasta > ~/temp.fas

all reads
 samtools view -b plate-10-GGAACGTGATTGCAGG.sort.bam | samtools fasta > ~/all-temp.fas

Can also align all the reads to siphateles.mtdna, setting this up and seeing what gets retained. 

(base) maccamp@farm:~/fish-lake/data/align-mtdna$ samtools view -F 4 -b plate-12-GGGAATCTGATGCAGG.fastq.sort.bam | samtools fasta

samtools view -F 4 -b plate-12-GGGAATCTGATGCAGG.fastq.sort.bam | samtools fasta >  plate-12-GGGAATCTGATGCAGG.fasta

(base) maccamp@farm:~/genomes/siphateles-mtdna$ blastn -query ~/fish-lake/data/align-mtdna/plate-12-GGGAATCTGATGCAGG.fasta -db siphateles.mtdna -outfmt 6

WORKS

(base) maccamp@farm:~/fish-lake/data/align-mtdna$ samtools view -F 4 -b plate-10-GGAACGTGATTGCAGG.fastq.sort.bam | samtools fasta
`>LH00444:71:2257TNLT4:6:2116:33167:7598/1
ACTCGTGATTACGTATAATTTTCTCCCCTTAAAAACACCCGTTATAATAATACCTGTTACCCTAAAACTAGCAGCCCTTATCGTGACAATCATCGGACTTTTAACCGCCATAGAACTTGCAGCCATAACTAATAA
>LH00444:71:2257TNLT4:6:2221:28077:27111/1
ACTCGTGATTACGTATAATTTTCTCCCCTTAAAAACACCCGTTATAATAATACCTGTTACCCTAAAACTAGCAGCCCTTATCGTGACAATCATCGGACTTTTAACCGCCATAGAACTTGCAGCCATAACTAATAA
>LH00444:71:2257TNLT4:7:1172:4318:2373/1
ACTCGTGATTACGTATAATTTTCTCCCCTTAAAAACACCCGTTATAATAATACCTGTTACCCTAAAACTAGCAGCCCTTATCGTGACAATCATCGGACTTTTAACCGCCATAGAACTTGCAGCCATAACTAATAA
>LH00444:71:2257TNLT4:6:2116:33167:7598/2
AGCCTAGTATATTTGAGAAGTGGTGTGTAGAAGTCGTAGGGGTGATTTTAATCTGTTTATTAGTTATGGCTGCAAGTTCTATGGCGGTTAAAAGTCCGATGATTGTCACGATAAGGGCTGCTAGTTTTAGGGTAACAGGTATTATTATAAC
>LH00444:71:2257TNLT4:6:2221:28077:27111/2
AGCCTAGTATATTTGAGAAGTGGTGTGTAGAAGTCGTAGGGGTGATTTTAATCTGTTTATTAGTTATGGCTGCAAGTTCTATGGCGGTTAAAAGTCCGATGATTGTCACGATAAGGGCTGCTAGTTTTAGGGTAACAGGTATTATTATAAC
>LH00444:71:2257TNLT4:7:1172:4318:2373/2
AGCCTAGTATATTTGAGAAGTGGTGTGTAGAAGTCGTAGGGGTGATTTTAATCTGTTTATTAGTTATGGCTGCAAGTTCTCTGGCGGTTAAAAGTCCGATGATTGTCACGATAAGGGCTGCTAGTTTTAGGGTAACAGGTATTATTATAAC`

93% similar to Gila robusta, G. cypha, G. nigrescens mtdna

830_1, 831_1

```{r}
million %>% filter(Sample %in% c("830_1","831_1"))
```
samtools view -F 4 -b plate-14-GGAAGACGGATGCAGG.fastq.sort.bam | samtools fasta >  plate-14-GGAAGACGGATGCAGG.fasta
samtools view -F 4 -b plate-14-GGATCCTGTATGCAGG.fastq.sort.bam | samtools fasta >  plate-14-GGATCCTGTATGCAGG.fasta

These seqs are 98-100% similar to Pogonichthys macrolepidotus!!!!
>LH00444:71:2257TNLT4:6:1202:25463:2191/2
GGGGCTTCTCAGTAGACAACGCAACACTAACGCGATTCTTCGCCTTCCACTTCCTATTCCCCTTCGTTATCGCCGGCGCAACCGTCCTACACTCAGCTGAGCGACGGCCCCCCCCGGGTCTCAGCCAGGGGTCCTGCATCCGTCTTCCAGA
>LH00444:71:2257TNLT4:7:1118:46486:18104/2
GGGGCTTCTCAGTAGACAACGCAACACTAACGCGATTCTTCGCCTTCCACTTCCTATTCCCCTTCGTTATCGCCGGCGCAACCGTCCTACACTCAGCTGAGCGACGGCCCCCCCCGGGTCTCAGCCAGGGGTCCTGCATCCGTCTTCCAGA
>LH00444:71:2257TNLT4:6:1202:25463:2191/1
ACCCCTGGCTGAGACCCGGGGGGGGCCGTCGCTCAGCTGAGTGTAGGACGGTTGCGCCGGCGATAACGAAGGGGAATAGGAAGTGGAAGGCGAAGAATCGCGTTAGTGTTGCGTTGTCTACTGAGAAGCCCCAGA
>LH00444:71:2257TNLT4:7:1118:46486:18104/1
ACCCCTGGCTGAGACCCGGGGGGGGCCGTCGCTCAGCTGAGTGTAGGACGGTTGCGCCGGCGATAACGAAGGGGAATAGGAAGTGGAAGGCGAAGAATCGCGTTAGTGTTGCGTTGTCTACTGAGAAGCCCCAGA
>LH00444:71:2257TNLT4:6:1256:20640:24043/1
CACGCGCATTTTCATTTTGAGTTCATCAACATTAAACGAAGGTAGGAAATGTGTGCGTTGACCACTGTTGTAATGATTCTGGTGGAAGTTTTGCACTTGATTGAGTAATATGTAATTCTTAAGTTATGTCTTAGA
>LH00444:71:2257TNLT4:6:1256:20640:24043/2

How about 780_1, 781_1

```{r}
million %>% filter(Sample %in% c("780_1","781_1"))
```


plate-12-GGAGTGGTCATGCAGG
samtools view -F 4 -b plate-12-GGAGTGGTCATGCAGG.fastq.sort.bam | samtools fasta >  plate-12-GGAGTGGTCATGCAGG.fasta

These are 100% identical to Siphateles boraxobius/alvordensis
LH00444:71:2257TNLT4:6:2172:29663:8424/2
LH00444:71:2257TNLT4:6:2172:29671:8438/2
LH00444:71:2257TNLT4:6:2209:49521:11562/2


Get some more weirdos 
```{r}
million %>% filter(Sample %in% c("335_1","42_1"))
```

plate-2-GGCGAACTTATGCAGG

samtools view -F 4 -b plate-2-GGCGAACTTATGCAGG.fastq.sort.bam | samtools fasta >  plate-2-GGCGAACTTATGCAGG.fasta
Pulled 0 reads, drop.
 
```{r}
df<-relocate(million, Sample) %>% rename(label=Sample)

t<-read.nexus(file = "outputs/103/nj-01.tre")
#t<-read.tree("outputs/103/pruned-reducedNew-min5-01.min4.phy.varsites.phy.contree")
#t<-midpoint_root(t) 
t<-root(t, outgroup = c("617_1","634_1","620_1","632_1","631_1"))

ggtree(t) %<+% df +
  
  geom_tippoint(aes(fill=Species), pch=21) +
  geom_tiplab(aes(color=Species, x=3400), size=.75) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  geom_treescale() +
  scale_color_viridis_d(option="H") +
 # geom_nodelab() +
  xlim(0,3600)

ggsave("outputs/103/nj-tree-inds.pdf", height=28, width = 10)

ggtree(t) %<+% df +
  
  geom_tippoint(aes(fill=Species), pch=21) +
  geom_tiplab(aes(label=paste0(Species," ",Vicinity," ",`Tributary/Collection Location`), color=Species, x=3400), size=.75) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  scale_color_viridis_d(option="H") +
  geom_treescale() +
 # geom_nodelab() +
  xlim(0,3600)

ggsave("outputs/103/nj-tree.pdf", height=28, width = 10)
```

```{sh, eval=FALSE}
#relaxing requirements for bootstrap correlation to 0.9 from 0.99 http://www.iqtree.org/doc/Command-Reference
#WARNING: bootstrap analysis did not converge. You should rerun with higher number of iterations (-nm option)
#Should run with nm for longer, this  converged!
iqtree2 -s pruned-01.min4.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo
iqtree2 -s pruned-01.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 1000 -nm 2000 -bcor 0.9 --redo

iqtree2 -s pruned-reducedNew-min5-01.min4.phy -T AUTO -st DNA -m GTR+ASC -bb 1000 --redo
iqtree2 -s pruned-reducedNew-min5-01.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 10000 --redo


#relaxing requirements for bootstrap correlation to 0.9 from 0.99 http://www.iqtree.org/doc/Command-Reference
iqtree2 -s pruned-reducedNew-min2-01.min4.phy -T AUTO -st DNA -m GTR+ASC -bb 1000 --redo
iqtree2 -s pruned-reducedNew-min2-01.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 1000 -bcor 0.9 --redo

```

165 inds, 4217 snps
265 6756 snps
921 inds 16048 sites

checked missing data with remove-missing.pl and didn't pull any out.  

```{r}
df<-relocate(million, Sample) %>% rename(label=Sample)
ogs<-c("617_1","620_1","631_1","632_1","634_1")
s
ggsave("outputs/103/ml-tree.pdf", height=24, width = 8.5)
```
```{r}
ggtree(t) %<+% df +
  
  geom_tippoint(aes(fill=Species), pch=21) +
  geom_tiplab(aes(x=.11, color=Species), size=.75) +
  scale_fill_viridis_d(option="H") +
  scale_color_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  geom_treescale() +
 # geom_nodelab() +
  xlim(0,.15)

ggsave("outputs/103/ml-tree-inds.pdf", height=24, width = 8.5)
```

What about boot?  

Read in SNPs 

```{r}
vcf<-read.vcfR("outputs/103/pruned-reducedNew-min5-01.vcf")
obj<-vcfR2genind(vcf)
obj@pop<-as.factor(min5$Vicinity)
obj@other<-min5
```

```{r}
t1<-aboot(obj, tree = "nj", distance = "nei.dist", sample = 100,
  cutoff = 50, showtree = TRUE, missing = "mean",
  strata = obj@other$`Tributary/Collection Location`)
```
```{r}
df2<-min5 %>% group_by(`Tributary/Collection Location`, Species, Vicinity) %>% summarize(Count=n())
df2
```

```{r}
t2<-midpoint_root(t1) 
ggtree(t2) %<+% df2 +
  geom_tippoint(aes(fill=Species), pch=21) +
  geom_tiplab(aes(label=paste0(Species," ",Vicinity," ",label)), size=1) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  geom_treescale() 

ggsave("outputs/103/pop-tree.pdf")
```


## Outgroup data
Relictus??? 

Let me see about that.  
