---
title: "104-obesus-pop-gen"
output: html_document
date: "2024-04-10"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message=FALSE)
```

The Fish Lake Valley fish appear to be the most divergent 'obesus' (not considering snyderi,toikona subspp.).   

```{r}
library(tidyverse)
library(ggrepel)
library(ggpubr)
```

Meta

Downsampled Paths are at 2e6.   

Don't have Alkali lake fish here. 

```{r}
mo5<-read_csv("meta/obesus-500k-meta.csv") %>% mutate(Path2=ifelse(Dedup > 2e6, paste0("data/downsample/",Ind,".reduced.bam"),
                     Path)) %>% mutate(Taxon=Species) %>% filter(Species=="obesus")

obs5<-read_csv("meta/obesus-500k-meta.csv") %>% mutate(Path2=ifelse(Dedup > 2e6, paste0("data/downsample/",Ind,".reduced.bam"),
                     Path)) %>% mutate(Taxon=Species) %>% filter(Species %in% c("obesus","snyderi","toikona"))
mo5 %>% group_by(Species, Vicinity) %>% summarize(Count=n())
```
337 inds without snyderi/toikona
381 inds   
474 inds with snyderi/toikona

```{r}
mo5 %>% select(Path) %>% write_csv("bamlists/mo5.bamlist", col_names = FALSE)
mo5 %>% select(Path2) %>% write_csv("bamlists/mo5-downsample.bamlist", col_names = FALSE)

obs5 %>% select(Path) %>% write_csv("bamlists/obs5.bamlist", col_names = FALSE)
obs5 %>% select(Path2) %>% write_csv("bamlists/obs5-downsample.bamlist", col_names = FALSE)
```

at ?? thresh.  
```{sh, eval=FALSE}
srun -p bigmemh -t 36:00:00 --mem=128G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/mo5.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-minInd 303 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.01 \
-out outputs/104/mo5-ibs-90 >outputs/104/mo5-ibs-90.out 2> outputs/104/mo5-ibs-90.err &

srun -p bigmemh -t 36:00:00 --mem=128G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/mo5-downsample.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-minInd 303 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.01 \
-out outputs/104/mo5-ibs-90-down >outputs/104/mo5-ibs-90-down.out 2> outputs/104/mo5-ibs-90-down.err &

srun -p high -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/mo5-downsample.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-minInd 303 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.01 \
-out outputs/104/mo5-ibs-90-high >outputs/104/mo5-ibs-90-high.out 2> outputs/104/mo5-ibs-90-high.err &

srun -p high -t 04:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/mo5-ibs-90-high.beagle.gz -K 2 -minMaf 0.01 -o outputs/104/mo5-K2 &
srun -p high -t 05:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/mo5-ibs-90-high.beagle.gz -K 3 -minMaf 0.01 -o outputs/104/mo5-K3 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/mo5-ibs-90-high.beagle.gz -K 4 -minMaf 0.01 -o outputs/104/mo5-K4 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/mo5-ibs-90-high.beagle.gz -K 5 -minMaf 0.01 -o outputs/104/mo5-K5 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/mo5-ibs-90-high.beagle.gz -K 6 -minMaf 0.01 -o outputs/104/mo5-K6 &

```

at maf of 0.01, 184142 sites
minMaf 0.05 90693 sites/336 inds
99948/381 inds at 0.05



Let's plot ngsAdmix


```{r, echo=FALSE}
PlotAdmix<- function(file, meta) {

q<-read_delim(file, delim=" ", col_names=FALSE)
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Last col empty
q<-q[1:length(colnames(q))-1]

#Bind met and arrange 
df<-bind_cols(q, meta) %>% mutate(Region=Vicinity) %>% arrange(Region) %>% mutate(Index=1:n())
df$Region<-factor(df$Region, levels=unique(df$Region))

rdf<-df %>% dplyr::select(Region, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 3:(3+length(colnames(q))-1))

#Make names for structure-like plot
labels<-rdf %>% group_by(Region) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Region,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  ylab("Q\n") +
  theme(legend.position = "") +
  theme(axis.text = element_text(size=10, face="bold")) +
  theme(axis.title.y=element_text(size=14, face="bold")) +
  scale_x_continuous(breaks=labels$Position, labels=labels$Region) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  #theme(axis.text.x=element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  theme(axis.ticks.x = element_blank())

}
```


```{r}
PlotAdmix("outputs/104/mo5-K2.qopt", mo5)
```

```{r}
PlotAdmix("outputs/104/mo5-K3.qopt", mo5)

ggsave("outputs/104/obesus-k3.pdf")
```


##Including snyderi/toikona
```{sh, eval=FALSE}
srun -p high -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/obs5-downsample.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-minInd 426 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.01 \
-out outputs/104/obs5-ibs-90-high >outputs/104/obs5-ibs-90-high.out 2> outputs/104/obs5-ibs-90-high.err &


srun -p high -t 04:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/obs5-ibs-90-high.beagle.gz -K 2 -minMaf 0.01 -o outputs/104/obs5-K2 &
srun -p high -t 05:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/obs5-ibs-90-high.beagle.gz -K 3 -minMaf 0.01 -o outputs/104/obs5-K3 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/obs5-ibs-90-high.beagle.gz -K 4 -minMaf 0.01 -o outputs/104/obs5-K4 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/obs5-ibs-90-high.beagle.gz -K 5 -minMaf 0.01 -o outputs/104/obs5-K5 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/obs5-ibs-90-high.beagle.gz -K 6 -minMaf 0.01 -o outputs/104/obs5-K6 &
```

```{r}
m <- as.matrix(read.table("outputs/104/mo5-ibs-90-high.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)

vardf<-as_data_frame(var) %>% mutate(x=1:n())
vardf$cumvar<-cumvar

```

```{r}
eigs<-ggplot(vardf %>% filter(x < 11)) +
  geom_col(aes(x=x, y=value))  +
  scale_x_continuous(breaks=c(1,5,10)) +
  xlab("\nPrincipal Component") +
  ylab("Eigenvalues\n") +
  theme_bw() +
  theme(axis.title = element_text(size=14, face="bold")) +
  theme(panel.grid=element_blank()) +
  theme(axis.text = element_text(size=12))

eigs
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(mo5)

labels12<- covs %>% select(V1, V2, Vicinity) %>% group_by(Vicinity) %>% summarize(x=mean(V1), y=mean(V2))

#labels12<-labels12 %>% filter(Vicinity=="Fish Lake Valley")


pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Vicinity), pch=21, cex=3, alpha=0.75) +
  geom_text_repel(data=labels12, aes(x=x, y=y, label=Vicinity), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("Genome-Wide PCA of obesus lineages") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) 

pc12
```

I would calle the Hot Creek Valley, Railroad Valley, and Little Fish Lake valley a "Tonopah" group

```{r}
pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Vicinity), pch=21, cex=3, alpha=0.75) +
  #geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("Genome-Wide PCA of obesus lineages") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) 

pc13
```


```{r}
ggarrange(pc12, eigs, widths=c(3,1))
ggsave("outputs/104/obesus-lineages-pca.pdf", width=12, height=5)
```

# With snyderi/toikona
```{r}
m <- as.matrix(read.table("outputs/104/obs5-ibs-90-high.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)

vardf<-as_data_frame(var) %>% mutate(x=1:n())
vardf$cumvar<-cumvar

```

```{r}
eigs<-ggplot(vardf %>% filter(x < 11)) +
  geom_col(aes(x=x, y=value))  +
  scale_x_continuous(breaks=c(1,5,10)) +
  xlab("\nPrincipal Component") +
  ylab("Eigenvalues\n") +
  theme_bw() +
  theme(axis.title = element_text(size=14, face="bold")) +
  theme(panel.grid=element_blank()) +
  theme(axis.text = element_text(size=12))

eigs
```


```{r}
covs<-eig$vectors[,1:4] %>% as_tibble() %>% bind_cols(obs5)

labels12<- covs %>% select(V1, V2, Vicinity) %>% group_by(Vicinity) %>% summarize(x=mean(V1), y=mean(V2))

#labels12<-labels12 %>% filter(Vicinity=="Fish Lake Valley")


pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Vicinity), pch=21, cex=3, alpha=0.75) +
  geom_text_repel(data=labels12, aes(x=x, y=y, label=Vicinity), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("PCs 1&2") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  theme(legend.position = "none")

pc12
```

````{r}
labels13<- covs %>% select(V1, V3, Vicinity) %>% group_by(Vicinity) %>% summarize(x=mean(V1), y=mean(V3))

pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Vicinity), pch=21, cex=3, alpha=0.75) +
  geom_text_repel(data=labels13, aes(x=x, y=y, label=Vicinity), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("PCs 1&3") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  theme(legend.position = "none")

pc13
```


````{r}
labels14<- covs %>% select(V1, V4, Vicinity) %>% group_by(Vicinity) %>% summarize(x=mean(V1), y=mean(V4))

pc14<-ggplot(covs) +
  geom_point(aes(x=V1, y=V4, fill=Vicinity), pch=21, cex=3, alpha=0.75) +
  geom_text_repel(data=labels14, aes(x=x, y=y, label=Vicinity), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC4", " ", round((100*var[4]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("PCs 1&4") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  theme(legend.position = "none")

pc14
```
```{r}
ggarrange(eigs, pc12, pc13, pc14)

ggsave("outputs/104/obesus-all-inds-500k-pca.pdf", width=12, height=12)
```

It looks like the Bridgeport samples may have some Owens Valley influence. Hmm...

Admixture plots follow:     

```{r}
PlotAdmix("outputs/104/obs5-K2.qopt", obs5)
PlotAdmix("outputs/104/obs5-K3.qopt", obs5)
PlotAdmix("outputs/104/obs5-K4.qopt", obs5)
```

```{r}
PlotAdmix("outputs/104/obs5-K5.qopt", obs5)

ggsave("outputs/104/obesus-snyder-toikona-k5.pdf")
```

```{r}
PlotAdmix("outputs/104/obs5-K6.qopt", obs5)

ggsave("outputs/104/obesus-snyder-toikona-k6.pdf")
```

## Get some owens fish for mtDNA sequencing

```{r, eval=TRUE}
file<-"~/github/fish-lake/outputs/104/obs5-K5.qopt"

q<-read_delim(file, delim=" ", col_names=FALSE)   

meta<-obs5
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Last col empty
q<-q[1:length(colnames(q))-1]

#Bind met and arrange 
df<-bind_cols(q, meta) %>% mutate(Region=Vicinity) %>% arrange(Region) %>% mutate(Index=1:n())
df$Region<-factor(df$Region, levels=unique(df$Region))

rdf<-df %>% dplyr::select(Region, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 3:(3+length(colnames(q))-1)) %>% filter(Region %in% c("Mammoth Mountain","Owens River", "Newark Valley"))

#Make names for structure-like plot
labels<-rdf %>% group_by(Region) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Region,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  ylab("Q\n") +
 # theme(legend.position = "") +
  theme(axis.text = element_text(size=10, face="bold")) +
  theme(axis.title.y=element_text(size=14, face="bold")) +
  scale_x_continuous(breaks=labels$Position, labels=labels$Region) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  #theme(axis.text.x=element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  theme(axis.ticks.x = element_blank())

```

```{r}
toikona<-df %>% filter(Region=="Owens River") %>% filter(Q4 > 0.99) 
snyderi<-df %>% filter(Region=="Owens River") %>% filter(Q5 > 0.99)

candidate<-bind_rows(snyderi,toikona)
write_csv(candidate, file="meta/candiate-mtDNA.csv")
```

Extract mtDNA....

First a candidate toikona:

samtools view -F 4 -b data/align-mtdna/plate-11-GGCGAACTTATGCAGG.fastq.sort.bam | samtools fasta > outputs/104/plate-11-GGCGAACTTATGCAGG.fasta

[M::bam2fq_mainloop] discarded 0 singletons
[M::bam2fq_mainloop] processed 63 reads

blastn -query outputs/104/plate-11-GGCGAACTTATGCAGG.fasta -db /home/maccamp/genomes/siphateles-mtdna/siphateles.mtdna -outfmt 6

NC_066943.1 Siphateles bicolor mitochondrion, complete genome is siphateles.mtdna

Comparing to 
AF237753.1	Siphateles bicolor obesa	Siphateles	bicolor	obesa	NA	Nevada, Churchill County, Toulon Drain
AF237752.1	Siphateles bicolor obesa	Siphateles	bicolor	obesa	NA	Nevada, Elko County, South Fork Humboldt River

blastn -query outputs/104/plate-11-GGCGAACTTATGCAGG.fasta -db /home/maccamp/genomes/siphateles-mtdna/s-obesa.mtdna -outfmt 6

let's try a snyderi
data/align/plate-11-GGAGTGGTCATGCAGG.sort.flt.bam

samtools view -F 4 -b data/align-mtdna/plate-11-GGAGTGGTCATGCAGG.fastq.sort.flt.bam | samtools fasta > outputs/104/plate-11-GGAGTGGTCATGCAGG.fasta

nomtdna

samtools view -F 4 -b data/align-mtdna/plate-11-GGCTGGCATATGCAGG.fastq.sort.flt.bam | samtools fasta > outputs/104/plate-11-GGCTGGCATATGCAGG.fasta

nomtdna
samtools view -F 4 data/align-mtdna/plate-11-GGCCAGTTCATGCAGG.fastq.sort.flt.bam


Get all the reads:
 cat meta/candiate-mtDNA.csv  | cut -f 25 -d ',' | perl -pe 's/.sort/.fastq.sort/g' | perl -pe 's/.flt//g' | perl -pe 's/align/align-mtdna/g' | while read line; do samtools view -F 4 -b $line | samtools fasta >> outputs/104/test.mtdna ; done;


looks to be a winner

