---
title: "104-obesus-pop-gen"
output: html_document
date: "2024-04-10"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message=FALSE)
```

The Fish Lake Valley fish appear to be the most divergent 'obesus' (not considering snyderi,toikona subspp.).   

```{r}
library(tidyverse)
library(ggrepel)
library(ggpubr)
```

Meta

Downsampled Paths are at 2e6.   

Don't have Alkali lake fish here. 

```{r}
mo5<-read_csv("meta/obesus-500k-meta.csv") %>% mutate(Path2=ifelse(Dedup > 2e6, paste0("data/downsample/",Ind,".reduced.bam"),
                     Path)) %>% mutate(Taxon=Species) %>% filter(Species=="obesus")

obs5<-read_csv("meta/obesus-500k-meta.csv") %>% mutate(Path2=ifelse(Dedup > 2e6, paste0("data/downsample/",Ind,".reduced.bam"),
                     Path)) %>% mutate(Taxon=Species) %>% filter(Species %in% c("obesus","snyderi","toikona"))
mo5 %>% group_by(Species, Vicinity) %>% summarize(Count=n())
```
337 inds without snyderi/toikona
381 inds   
474 inds with snyderi/toikona

```{r}
mo5 %>% select(Path) %>% write_csv("bamlists/mo5.bamlist", col_names = FALSE)
mo5 %>% select(Path2) %>% write_csv("bamlists/mo5-downsample.bamlist", col_names = FALSE)

obs5 %>% select(Path) %>% write_csv("bamlists/obs5.bamlist", col_names = FALSE)
obs5 %>% select(Path2) %>% write_csv("bamlists/obs5-downsample.bamlist", col_names = FALSE)
```

at ?? thresh.  
```{sh, eval=FALSE}
srun -p bigmemh -t 36:00:00 --mem=128G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/mo5.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-minInd 303 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.01 \
-out outputs/104/mo5-ibs-90 >outputs/104/mo5-ibs-90.out 2> outputs/104/mo5-ibs-90.err &

srun -p bigmemh -t 36:00:00 --mem=128G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/mo5-downsample.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-minInd 303 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.01 \
-out outputs/104/mo5-ibs-90-down >outputs/104/mo5-ibs-90-down.out 2> outputs/104/mo5-ibs-90-down.err &

srun -p high -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/mo5-downsample.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-minInd 303 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.01 \
-out outputs/104/mo5-ibs-90-high >outputs/104/mo5-ibs-90-high.out 2> outputs/104/mo5-ibs-90-high.err &

srun -p high -t 04:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/mo5-ibs-90-high.beagle.gz -K 2 -minMaf 0.01 -o outputs/104/mo5-K2 &
srun -p high -t 05:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/mo5-ibs-90-high.beagle.gz -K 3 -minMaf 0.01 -o outputs/104/mo5-K3 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/mo5-ibs-90-high.beagle.gz -K 4 -minMaf 0.01 -o outputs/104/mo5-K4 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/mo5-ibs-90-high.beagle.gz -K 5 -minMaf 0.01 -o outputs/104/mo5-K5 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/mo5-ibs-90-high.beagle.gz -K 6 -minMaf 0.01 -o outputs/104/mo5-K6 &

```

at maf of 0.01, 184142 sites
minMaf 0.05 90693 sites/336 inds
99948/381 inds at 0.05



Let's plot ngsAdmix


```{r, echo=FALSE}
PlotAdmix<- function(file, meta) {

q<-read_delim(file, delim=" ", col_names=FALSE)
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Last col empty
q<-q[1:length(colnames(q))-1]

#Bind met and arrange 
df<-bind_cols(q, meta) %>% mutate(Region=Vicinity) %>% arrange(Region) %>% mutate(Index=1:n())
df$Region<-factor(df$Region, levels=unique(df$Region))

rdf<-df %>% dplyr::select(Region, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 3:(3+length(colnames(q))-1))

#Make names for structure-like plot
labels<-rdf %>% group_by(Region) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Region,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  ylab("Q\n") +
  theme(legend.position = "") +
  theme(axis.text = element_text(size=10, face="bold")) +
  theme(axis.title.y=element_text(size=14, face="bold")) +
  scale_x_continuous(breaks=labels$Position, labels=labels$Region) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  #theme(axis.text.x=element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  theme(axis.ticks.x = element_blank())

}
```


```{r}
PlotAdmix("outputs/104/mo5-K2.qopt", mo5)
```

```{r}
PlotAdmix("outputs/104/mo5-K3.qopt", mo5)

ggsave("outputs/104/obesus-k3.pdf")
```


##Including snyderi/toikona
```{sh, eval=FALSE}
srun -p high -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/obs5-downsample.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-minInd 426 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.01 \
-out outputs/104/obs5-ibs-90-high >outputs/104/obs5-ibs-90-high.out 2> outputs/104/obs5-ibs-90-high.err &


srun -p high -t 04:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/obs5-ibs-90-high.beagle.gz -K 2 -minMaf 0.01 -o outputs/104/obs5-K2 &
srun -p high -t 05:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/obs5-ibs-90-high.beagle.gz -K 3 -minMaf 0.01 -o outputs/104/obs5-K3 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/obs5-ibs-90-high.beagle.gz -K 4 -minMaf 0.01 -o outputs/104/obs5-K4 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/obs5-ibs-90-high.beagle.gz -K 5 -minMaf 0.01 -o outputs/104/obs5-K5 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/104/obs5-ibs-90-high.beagle.gz -K 6 -minMaf 0.01 -o outputs/104/obs5-K6 &
```

```{r}
m <- as.matrix(read.table("outputs/104/mo5-ibs-90-high.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)

vardf<-as_data_frame(var) %>% mutate(x=1:n())
vardf$cumvar<-cumvar

```

```{r}
eigs<-ggplot(vardf %>% filter(x < 11)) +
  geom_col(aes(x=x, y=value))  +
  scale_x_continuous(breaks=c(1,5,10)) +
  xlab("\nPrincipal Component") +
  ylab("Eigenvalues\n") +
  theme_bw() +
  theme(axis.title = element_text(size=14, face="bold")) +
  theme(panel.grid=element_blank()) +
  theme(axis.text = element_text(size=12))

eigs
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(mo5)

labels12<- covs %>% select(V1, V2, Vicinity) %>% group_by(Vicinity) %>% summarize(x=mean(V1), y=mean(V2))

#labels12<-labels12 %>% filter(Vicinity=="Fish Lake Valley")


pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Vicinity), pch=21, cex=3, alpha=0.75) +
  geom_text_repel(data=labels12, aes(x=x, y=y, label=Vicinity), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("Genome-Wide PCA of obesus lineages") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) 

pc12
```

I would calle the Hot Creek Valley, Railroad Valley, and Little Fish Lake valley a "Tonopah" group

```{r}
pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Vicinity), pch=21, cex=3, alpha=0.75) +
  #geom_text_repel(data=text12, aes(x=x, y=y, label=Region), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("Genome-Wide PCA of obesus lineages") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) 

pc13
```


```{r}
ggarrange(pc12, eigs, widths=c(3,1))
ggsave("outputs/104/obesus-lineages-pca.pdf", width=12, height=5)
```

# With snyderi/toikona
```{r}
m <- as.matrix(read.table("outputs/104/obs5-ibs-90-high.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)

vardf<-as_data_frame(var) %>% mutate(x=1:n())
vardf$cumvar<-cumvar

```

```{r}
eigs<-ggplot(vardf %>% filter(x < 11)) +
  geom_col(aes(x=x, y=value))  +
  scale_x_continuous(breaks=c(1,5,10)) +
  xlab("\nPrincipal Component") +
  ylab("Eigenvalues\n") +
  theme_bw() +
  theme(axis.title = element_text(size=14, face="bold")) +
  theme(panel.grid=element_blank()) +
  theme(axis.text = element_text(size=12))

eigs
```


```{r}
covs<-eig$vectors[,1:4] %>% as_tibble() %>% bind_cols(obs5)

labels12<- covs %>% select(V1, V2, Vicinity) %>% group_by(Vicinity) %>% summarize(x=mean(V1), y=mean(V2))

#labels12<-labels12 %>% filter(Vicinity=="Fish Lake Valley")


pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Vicinity), pch=21, cex=3, alpha=0.75) +
  geom_text_repel(data=labels12, aes(x=x, y=y, label=Vicinity), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("PCs 1&2") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  theme(legend.position = "none")

pc12
```

````{r}
labels13<- covs %>% select(V1, V3, Vicinity) %>% group_by(Vicinity) %>% summarize(x=mean(V1), y=mean(V3))

pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Vicinity), pch=21, cex=3, alpha=0.75) +
  geom_text_repel(data=labels13, aes(x=x, y=y, label=Vicinity), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("PCs 1&3") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  theme(legend.position = "none")

pc13
```


````{r}
labels14<- covs %>% select(V1, V4, Vicinity) %>% group_by(Vicinity) %>% summarize(x=mean(V1), y=mean(V4))

pc14<-ggplot(covs) +
  geom_point(aes(x=V1, y=V4, fill=Vicinity), pch=21, cex=3, alpha=0.75) +
  geom_text_repel(data=labels14, aes(x=x, y=y, label=Vicinity), max.overlaps = Inf) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC4", " ", round((100*var[4]),2), "%", sep = "")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  ggtitle("PCs 1&4") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  theme(legend.position = "none")

pc14
```
```{r}
ggarrange(eigs, pc12, pc13, pc14)

ggsave("outputs/104/obesus-all-inds-500k-pca.pdf", width=12, height=12)
```

It looks like the Bridgeport samples may have some Owens Valley influence. Hmm...

Admixture plots follow:     

```{r}
PlotAdmix("outputs/104/obs5-K2.qopt", obs5)
PlotAdmix("outputs/104/obs5-K3.qopt", obs5)
PlotAdmix("outputs/104/obs5-K4.qopt", obs5)
```

```{r}
PlotAdmix("outputs/104/obs5-K5.qopt", obs5)

ggsave("outputs/104/obesus-snyder-toikona-k5.pdf")
```

```{r}
PlotAdmix("outputs/104/obs5-K6.qopt", obs5)

ggsave("outputs/104/obesus-snyder-toikona-k6.pdf")
```

## Get some owens fish for mtDNA sequencing

```{r, eval=TRUE}
file<-"~/github/fish-lake/outputs/104/obs5-K5.qopt"

q<-read_delim(file, delim=" ", col_names=FALSE)   

meta<-obs5
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Last col empty
q<-q[1:length(colnames(q))-1]

#Bind met and arrange 
df<-bind_cols(q, meta) %>% mutate(Region=Vicinity) %>% arrange(Region) %>% mutate(Index=1:n())
df$Region<-factor(df$Region, levels=unique(df$Region))

rdf<-df %>% dplyr::select(Region, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 3:(3+length(colnames(q))-1)) %>% filter(Region %in% c("Mammoth Mountain","Owens River", "Newark Valley"))

#Make names for structure-like plot
labels<-rdf %>% group_by(Region) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Region,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  ylab("Q\n") +
 # theme(legend.position = "") +
  theme(axis.text = element_text(size=10, face="bold")) +
  theme(axis.title.y=element_text(size=14, face="bold")) +
  scale_x_continuous(breaks=labels$Position, labels=labels$Region) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  #theme(axis.text.x=element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  theme(axis.ticks.x = element_blank())

```

```{r}
toikona<-df %>% filter(Region=="Owens River") %>% filter(Q4 > 0.99) 
snyderi<-df %>% filter(Region=="Owens River") %>% filter(Q5 > 0.99)

candidate<-bind_rows(snyderi,toikona)
write_csv(candidate, file="meta/candiate-mtDNA.csv")
```

```{r}
candidate %>% dplyr::select(Q4, Q5, GVL_Code, Vicinity, `Tributary/Collection Location`, Species)
```
Extract mtDNA....

First a candidate toikona:

samtools view -F 4 -b data/align-mtdna/plate-11-GGCGAACTTATGCAGG.fastq.sort.bam | samtools fasta > outputs/104/plate-11-GGCGAACTTATGCAGG.fasta

[M::bam2fq_mainloop] discarded 0 singletons
[M::bam2fq_mainloop] processed 63 reads

blastn -query outputs/104/plate-11-GGCGAACTTATGCAGG.fasta -db /home/maccamp/genomes/siphateles-mtdna/siphateles.mtdna -outfmt 6

NC_066943.1 Siphateles bicolor mitochondrion, complete genome is siphateles.mtdna

Comparing to 
AF237753.1	Siphateles bicolor obesa	Siphateles	bicolor	obesa	NA	Nevada, Churchill County, Toulon Drain
AF237752.1	Siphateles bicolor obesa	Siphateles	bicolor	obesa	NA	Nevada, Elko County, South Fork Humboldt River

blastn -query outputs/104/plate-11-GGCGAACTTATGCAGG.fasta -db /home/maccamp/genomes/siphateles-mtdna/s-obesa.mtdna -outfmt 6

let's try a snyderi
data/align/plate-11-GGAGTGGTCATGCAGG.sort.flt.bam

samtools view -F 4 -b data/align-mtdna/plate-11-GGAGTGGTCATGCAGG.fastq.sort.flt.bam | samtools fasta > outputs/104/plate-11-GGAGTGGTCATGCAGG.fasta

nomtdna

samtools view -F 4 -b data/align-mtdna/plate-11-GGCTGGCATATGCAGG.fastq.sort.flt.bam | samtools fasta > outputs/104/plate-11-GGCTGGCATATGCAGG.fasta

nomtdna
samtools view -F 4 data/align-mtdna/plate-11-GGCCAGTTCATGCAGG.fastq.sort.flt.bam


Get all the reads:
 cat meta/candiate-mtDNA.csv  | cut -f 25 -d ',' | perl -pe 's/.sort/.fastq.sort/g' | perl -pe 's/.flt//g' | perl -pe 's/align/align-mtdna/g' | while read line; do samtools view -F 4 -b $line | samtools fasta >> outputs/104/test.mtdna ; done;
 
 

looks to be a winner

blastn -query test.mtdna -db OL457405.1.fasta -outfmt 6
(base) ➜  104 git:(main) ✗ blastn -query test.mtdna -db OL457405.1.fasta -outfmt 6 | awk '$4 > 140 {print}'

(base) ➜  104 git:(main) ✗ blastn -query test.mtdna -db AF370056.1.fasta -outfmt 6 | awk '$4 > 140 {print}'

LH00444:71:2257TNLT4:6:2294:14287:15232/2	AF370056.1	100.000	151	0	0	1	151	226	76	8.34e-80	279
LH00444:71:2257TNLT4:7:1375:32341:3227/2	AF370056.1	98.611	144	2	0	7	150	220	77	1.41e-72	255

blastn -query test.mtdna -db AF370056.1.fasta -outfmt 6 | awk '$4 > 120 {print}' | cut -f 1-4
(base) ➜  104 git:(main) ✗ blastn -query test.mtdna -db AF370056.1.fasta -outfmt 6 | awk '$4 > 120 {print}' | cut -f 1-4

LH00444:71:2257TNLT4:6:2294:14287:15232/1	AF370056.1	100.000	128
LH00444:71:2257TNLT4:7:1375:32341:3227/1	AF370056.1	97.656	128
LH00444:71:2257TNLT4:6:2294:14287:15232/2	AF370056.1	100.000	151
LH00444:71:2257TNLT4:7:1375:32341:3227/2	AF370056.1	98.611	144

blastn -query test.mtdna -db AF370056.1.fasta -outfmt 6 | awk '$4 > 75 {print}' | cut -f 1-4
>LH00444:71:2257TNLT4:7:1375:32341:3227/1
AACAATAATGGCAAGCCTACGAAAAACCCACCCACTAATAAAAATCGCTAACGATGCACTAGTTGACCTCCCCACGCCATCTAATATTTCCGTAATATGAAACTTTGGATCTCTCCTAGGATTATGTTTCATTAC
>LH00444:71:2257TNLT4:7:1375:32341:3227/2
CACAACTAACGCCTCGGCAGATGTGGGTTACTGATGAAAATGCAGTTGAGATATCAGAGGTGTAGTGTATGGCTAAAAACAGCCCTGTCAGAATTTGGGTAATTAAACATAATCCTAGGAGAGATCCAAAGTTTCATATTACTGAAATATC


75 bases
(base) ➜  104 git:(main) ✗ blastn -query test.mtdna -db AF370056.1.fasta -outfmt 6 | awk '$4 > 75 {print}' | cut -f 1-4

LH00444:71:2257TNLT4:7:1385:37415:1504/2	AF370056.1	98.718	78
LH00444:71:2257TNLT4:7:2327:8793:24827/2	AF370056.1	98.718	78
LH00444:71:2257TNLT4:6:1115:15687:11156/2	AF370056.1	100.000	93
LH00444:71:2257TNLT4:6:1408:22946:12514/2	AF370056.1	100.000	93
LH00444:71:2257TNLT4:7:2227:10314:9447/2	AF370056.1	100.000	104
LH00444:71:2257TNLT4:7:1266:35036:9377/2	AF370056.1	100.000	110
LH00444:71:2257TNLT4:7:1348:2861:3914/2	AF370056.1	98.182	110
LH00444:71:2257TNLT4:6:2294:14287:15232/1	AF370056.1	100.000	128
LH00444:71:2257TNLT4:7:1375:32341:3227/1	AF370056.1	97.656	128
LH00444:71:2257TNLT4:6:2294:14287:15232/2	AF370056.1	100.000	151
LH00444:71:2257TNLT4:7:1375:32341:3227/2	AF370056.1	98.611	144


(base) maccamp@farm:~/fish-lake/outputs/104/mtdna$ grep "LH00444:71:2257TNLT4:7:1385:37415:1504/2" *.fasta
plate-12-GGCGCATACATGCAGG.fasta:>LH00444:71:2257TNLT4:7:1385:37415:1504/2


(base) maccamp@farm:~/fish-lake/data/align-mtdna$ for f in *.fastq; do echo $f; grep "LH00444:71:2257TNLT4:7:1375:32341:3227/1" $f; done;


cat meta/candiate-mtDNA.csv  | cut -f 25 -d ',' | perl -pe 's/.sort/.fastq.sort/g' | perl -pe 's/.sort.flt.bam//g' | perl -pe 's/align/align-mtdna/g' | while read line; do echo $line; grep "LH00444:71:2257TNLT4:6:2294:14287:15232/2" $line; done;

 cat meta/candiate-mtDNA.csv  | cut -f 25 -d ',' | perl -pe 's/.sort/.fastq.sort/g' | perl -pe 's/.flt//g' | perl -pe 's/align/align-mtdna/g' | while read line; do samtools view -F 4 -b $line | samtools fasta >> outputs/104/mtdna/$(basename $line .fastq.sort.bam).fasta ; done;
 
 Let's look for some fish!
 
 grep "LH00444:71:2257TNLT4:7:1385:37415:1504/2" *.fasta
plate-12-GGCGCATACATGCAGG.fasta:>LH00444:71:2257TNLT4:7:1385:37415:1504/2
(base) maccamp@farm:~/fish-lake/outputs/104/mtdna$ grep "LH00444:71:2257TNLT4:7:2327:8793:24827/2" *.fasta
plate-12-GGCGCATACATGCAGG.fasta:>LH00444:71:2257TNLT4:7:2327:8793:24827/2
(base) maccamp@farm:~/fish-lake/outputs/104/mtdna$ grep "LH00444:71:2257TNLT4:7:1348:2861:3914/2" *.fasta
plate-12-GGCAAGACTATGCAGG.fasta:>LH00444:71:2257TNLT4:7:1348:2861:3914/2
base) maccamp@farm:~/fish-lake/outputs/104/mtdna$ grep "LH00444:71:2257TNLT4:7:1375:32341:3227/1" *.fasta
plate-11-GGAGCCATGCTGCAGG.fasta:>LH00444:71:2257TNLT4:7:1375:32341:3227/1
(base) maccamp@farm:~/fish-lake/outputs/104/mtdna$ grep "LH00444:71:2257TNLT4:7:1375:32341:3227/2" *.fasta
plate-11-GGAGCCATGCTGCAGG.fasta:>LH00444:71:2257TNLT4:7:1375:32341:3227/2

plate-12-GGCGCATACATGCAGG.fasta SEP22_11
plate-12-GGCGCATACATGCAGG.fasta SEP22_11
plate-12-GGCAAGACTATGCAGG.fasta SWP22_19
plate-11-GGAGCCATGCTGCAGG.fasta MUS22_11
plate-11-GGAGCCATGCTGCAGG.fasta MUS22_11
 
9.9999999e-10,9.9999999e-10,9.9999999e-10,9.9999999e-10,0.999999996,plate-12-GGCGCATACATGCAGG,D6,SEP22_11,OWEN_SEPD_2022_011_L,Owens River,SE Pond (White Mountain Research Center),snyderi,NA,NA,2022,NA,TuiChub_RADseq_Plate12_L,NEBNext 21,GTTTCG,CGCATACA,12,4206859,1960769,0.199646,data/align/plate-12-GGCGCATACATGCAGG.sort.flt.bam,data/align/plate-12-GGCGCATACATGCAGG.sort.flt.bam,snyderi,Owens River,392

grep plate-12-GGCAAGACTATGCAGG candiate-mtDNA.csv 
9.9999999e-10,9.9999999e-10,9.9999999e-10,9.9999999e-10,0.999999996,plate-12-GGCAAGACTATGCAGG,C7,SWP22_19,OWEN_SWPD_2022_019_L,Owens River,SW Pond (White Mountain Research Center),snyderi,NA,NA,2022,NA,TuiChub_RADseq_Plate12_L,NEBNext 21,GTTTCG,CAAGACTA,12,6613191,2893706,0.295683,data/align/plate-12-GGCAAGACTATGCAGG.sort.flt.bam,data/downsample/plate-12-GGCAAGACTATGCAGG.reduced.bam,snyderi,Owens River,396
grep plate-11-GGAGCCATGCTGCAGG candiate-mtDNA.csv 
9.9999999e-10,9.9999999e-10,0.00128424319545781,0.998715753804542,9.9999999e-10,plate-11-GGAGCCATGCTGCAGG,G6,MUS22_11,OWEN_MULE_2022_011_K,Owens River,Mule Spring,toikona,NA,NA,2022,NA,TuiChub_RADseq_Plate11_K,NEBNext 20,GTGGCC,AGCCATGC,11,4768952,873301,0.0769861,data/align/plate-11-GGAGCCATGCTGCAGG.sort.flt.bam,data/align/plate-11-GGAGCCATGCTGCAGG.sort.flt.bam,toikona,Owens River,337


MUS22_11 seq overlaps the two...

Can I extract the cytb sequence from MUS22_11 ?

plate-12-GGCAAGACTATGCAGG.fastq.sort.bam is aligned to NC_066943.1 ~/genomes/siphateles-mtdna/siphateles.mtdna

Filter to aligned reads
```{sh, eval=FALSE}
samtools view -bF 2308 data/align-mtdna/plate-12-GGCAAGACTATGCAGG.fastq.sort.bam -o outputs/104/MUS22_11-filtered.bam
samtools index outputs/104/MUS22_11-filtered.bam
/home/maccamp/proovread/bin/bam2cns --bam  outputs/104/MUS22_11-filtered.bam --coverage 0.5 --ref /home/maccamp/genomes/siphateles-mtdna/siphateles.mtdna --prefix outputs/104/MUS22_11-mtdna
~/seqtk/seqtk seq -a -q20 -n N outputs/104/MUS22_11-mtdna.fq > outputs/104/MUS22_11-mtdna-q20.fasta
samtools faidx  outputs/104/MUS22_11-mtdna-q20.fasta NC_066943.1:14394-15534 > outputs/104/MUS22_11-cytb-q20.fasta
~/seqtk/seqtk seq -a -q 10 -n N outputs/104/MUS22_11-mtdna.fq > outputs/104/MUS22_11-mtdna-q10.fasta
samtools faidx  outputs/104/MUS22_11-mtdna-q10.fasta NC_066943.1:14394-15534 > outputs/104/MUS22_11-cytb-q10.fasta

~/seqtk/seqtk seq -a -q 05 -n N outputs/104/MUS22_11-mtdna.fq > outputs/104/MUS22_11-mtdna-q05.fasta
samtools faidx  outputs/104/MUS22_11-mtdna-q05.fasta NC_066943.1:14394-15534 > outputs/104/MUS22_11-cytb-q05.fasta
```


samtools view -bF 2308 data/align-mtdna/plate-12-GGCAAGACTATGCAGG.fastq.sort.bam -o outputs/104/MUS22_11-filtered.bam


q10 consensus is this:
>MUS22_11
GCCTATGCCATCCTACGATCTATCCCAAATAAACTAGGAGGGGTCCTCGCACTACTATTTAGTATCCTAGTACTGCTAGTGGTACCAATCCTACACACCTCAAAGCAAC

Something doesn't add up here. These seqs aren't in that sample?

Let's get the two seqs as fastqs
>LH00444:71:2257TNLT4:7:1375:32341:3227/1
AACAATAATGGCAAGCCTACGAAAAACCCACCCACTAATAAAAATCGCTAACGATGCACTAGTTGACCTCCCCACGCCATCTAATATTTCCGTAATATGAAACTTTGGATCTCTCCTAGGATTATGTTTCATTAC
>LH00444:71:2257TNLT4:7:1375:32341:3227/2
CACAACTAACGCCTCGGCAGATGTGGGTTACTGATGAAAATGCAGTTGAGATATCAGAGGTGTAGTGTATGGCTAAAAACAGCCCTGTCAGAATTTGGGTAATTAAACATAATCCTAGGAGAGATCCAAAGTTTCATATTACTGAAATATC

Getting separate files.

```{sh, eval=FALSE}
 cat meta/candiate-mtDNA.csv  | cut -f 25 -d ',' | perl -pe 's/.sort/.fastq.sort/g' | perl -pe 's/.flt//g' | perl -pe 's/align/align-mtdna/g' | while read line; do samtools view -bF 2308 $line > outputs/104/cytb/$(basename $line .fastq.sort.bam).bam; done;
 
for f in outputs/104/cytb/*bam; do samtools index $f; done;

for f in outputs/104/cytb/*.bam; do echo $f;  /home/maccamp/proovread/bin/bam2cns --bam  $f --coverage 1 --ref /home/maccamp/genomes/siphateles-mtdna/siphateles.mtdna --prefix outputs/104/cytb/$(basename $f .bam) ; done;

for f in outputs/104/cytb/*.fq; do ~/seqtk/seqtk seq -a -q 10 -n N $f > outputs/104/cytb/$(basename $f .fq).fasta; done;
for f in  outputs/104/cytb/*.fasta; do samtools faidx $f NC_066943.1:14394-15534 > outputs/104/cytb/$(basename $f .fasta).fna; done;
 ```
 
 (base) maccamp@farm:~/fish-lake/outputs/104/cytb$ for f in *.fna; do echo $f; cat $f; done;
 plate-11-GGAGCCATGCTGCAGG.fna
 
>NC_066943.1:14394-15534
ATGGCAAGCCTACGAAAAACCCACCCACTAATAAAAATCGCTAACGATGCACTAGTTGACCTCCCAACGCCATCTAATATTTCAGTAATATGAAACTTTGGATCTCTCCTAGGATTATGTTTAATTACCCAAATTCTGACAGGGCTATTTTTAGCCATACACTACACCTCTGATATCTCAACTGCATTTTCATCAGTAACCCACATCTGCCGAGACGTTAATTATGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN

This is MUS22_11
>MUS22_11
ATGGCAAGCCTACGAAAAACCCACCCACTAATAAAAATCGCTAACGATGCACTAGTTGACCTCCCAACGCCATCTAATATTTCAGTAATATGAAACTTTGGATCTCTCCTAGGATTATGTTTAATTACCCAAATTCTGACAGGGCTATTTTTAGCCATACACTACACCTCTGATATCTCAACTGCATTTTCATCAGTAACCCACATCTGCCGAGACGTTAATTATG

plate-12-GGCAAGACTATGCAGG.fna
>NC_066943.1:14394-15534
NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNGCCTATGCCATCCTACGATCTATCCCAAATAAACTAGGAGGGGTCCTCGCACTACTATTTAGTATCCTAGTACTGCTAGTGGTACCAATCCTACACACCTCAAAGCAACGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN

plate-12-GGCGCATACATGCAGG.fna
>NC_066943.1:14394-15534
NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNATTGGCCAAGCCGCATCAGTCCTATACTTTGCACTTTTCCTTATTCTTGTCCCACTGGCAGGGTGAGTAGAAAATAAAGCACTGAAATGAGCTT

>plate-11-GGAGCCATGCTGCAGG.
ATGGCAAGCCTACGAAAAACCCACCCACTAATAAAAATCGCTAACGATGCACTAGTTGACCTCCCAACGCCATCTAATATTTCAGTAATATGAAACTTTGGATCTCTCCTAGGATTATGTTTAATTACCCAAATTCTGACAGGGCTATTTTTAGCCATACACTACACCTCTGATATCTCAACTGCATTTTCATCAGTAACCCACATCTGCCGAGACGTTAATTATG
>plate-12-GGCAAGACTATGCAGG
GCCTATGCCATCCTACGATCTATCCCAAATAAACTAGGAGGGGTCCTCGCACTACTATTTAGTATCCTAGTACTGCTAGTGGTACCAATCCTACACACCTCAAAGCAACG
>PLATE-12-GGCGCATACATGCAGG
ATTGGCCAAGCCGCATCAGTCCTATACTTTGCACTTTTCCTTATTCTTGTCCCACTGGCAGGGTGAGTAG
AAAATAAAGCACTGAAATGAGCTT

How about like this?

samtools fastq plate-11-GGAGCCATGCTGCAGG.bam > plate-11-GGAGCCATGCTGCAGG.fastq

~/seqtk/seqtk seq -a -q 20 -n N plate-11-GGAGCCATGCTGCAGG.fastq >  plate-11-GGAGCCATGCTGCAGG-q20.fasta
>LH00444:71:2257TNLT4:6:1118:9966:16801/2
NAAGATGTTAAGATGGGCCCTGAGAAGCTCCGCATGTCTCCTCGACGNTCTCCCAAGTGGACGGCGCCGCCGAGAAGGGGTACAAAGCTNTCCCNCCAGTGGNGGAGGCGGTNGCAGCACANCTTTGCCCGCCCTCTGCGAGNTGGCGCTC
>LH00444:71:2257TNLT4:6:1142:5386:26606/2
GAAGATGTTAAGATGGGCCCTGAGAAGCTCCGCATGTCTCCTCGACGCTCTCCCAAGTGGACGGCGCCGCCGAGAAGGGGTACAAAGCTATCCCCCCAGTGGAGGAGGCGGTCGCAGCACACCTTTGCCCGCCCTCTGCGAGATGGCGCTC
>LH00444:71:2257TNLT4:6:2166:23472:20569/2
GAAGATGTTAAGATGGGCCCTGAGAAGCTCCGCATGTCTCCTCGACGCTCTCCCAAGTGGACGGCGCCGCCGAGAAGGGGTACAAAGCTATCCCCCCAGTGGAGGAGGCGGTCGCAGCACACCTTTGCCCGCCCTCTGCGAGATGGCGCTC
>LH00444:71:2257TNLT4:7:2156:42254:10469/2
GAAGATGTTAAGATGGGCCCTGAGAAGCTCCGCATGTCTCCTCGACGCTCTCCCAAGTGGACGGCGCCGCCGAGAAGGGGTACAAAGCTATCCCCCCAGTGGAGGAGGCGGTCGCAGCANACCTTTGCCCGCCCTCTGCGAGATGGCGCTC
>LH00444:71:2257TNLT4:7:2327:29485:11590/2
GAAGATGTTAAGATGGGCCCTGAGAAGCTCCGCATGTCTCCTCGACGCTCTCCCAAGTGGACGGCGCCGCCGAGAAGGGGTACAAAGCTATCCCCCCAGTGGAGGAGGCGGTCGCAGCACACCTTTGCCCGCCCTCTGCGAGATGGCGCTC
>LH00444:71:2257TNLT4:7:2166:28821:29156/1
CACAACCACGGGCACAGCAGCGTCTGTTAAATTATCATGTGGGAACGCAAAGTGGGAAAAATAAAACCTACAGCAAGACCTGGTGGGCATATAAAAAATTCAGACTTTAANCGAGACCAATGACTTGAAGAACCA
>LH00444:71:2257TNLT4:7:2166:28821:29156/2
AGTTGAATTACAACGGTGGTTCTTCAAGTCATTGGTNTCGGTTAAAGTCTGAATTTTTTATATGCCCACCAGGTCTTGCTGTAGGTTTTATTTTTCCCACTTTGCGTTCCCACATGATAATTTAACAGACGCTGCTGTGCCCGTGGTTGTG
>LH00444:71:2257TNLT4:6:2294:14287:15232/1
AACAATAATGGCAAGCCTACGAAAAACCCACCCACTAATAAAAATCGCTAACGATGCACTAGTTGACCTCCCAACGCCATCTAATATTTCAGTAATATGAAACTTTGGATCTCTCCTAGGATTATGTTTAATTAC
>LH00444:71:2257TNLT4:7:1375:32341:3227/1
AACAATAATGGCAAGCCTACGAAAAACCCACCCACTAATAAAAATCGCTAACGATGCACTAGTTGACCTCCCNACGCCATCTAATATTTCNGTAATATGAAACTTTGGATCTCTCCTNGGATTATGTTTNATTAC
>LH00444:71:2257TNLT4:6:2294:14287:15232/2
CATAATTAACGTCTCGGCAGATGTGGGTTACTGATGAAAATGCAGTTGAGATATCAGAGGTGTAGTGTATGGCTAAAAATAGCCCTGTCAGAATTTGGGTAATTAAACATAATCCTANGAGAGATCCAAAGTTTCATATTACTGAAATATT
>LH00444:71:2257TNLT4:7:1375:32341:3227/2
CANAANNAACNNCNCNGCANANGTNGNNNACTNATNAAAATGCAGNTNAGATNTCAGAGGTGNAGTGTATGGCTAAANANAGCCCTNTCAGAATTTGGGTAATTAAACNTAATCCTAGGAGAGATCCNAAGTTTCATATTACTGAANNATN
>LH00444:71:2257TNLT4:6:1123:43792:12851/2
GATGAAAATGCAGTTGAGATATCAGAGGTGTAGTGTAGCTGTCTGTGGAGAGTACAGAAAGTGCTCAAAAAATCTTCTTTTTTTTCAGAAGATGAACGAAAGCGAGGGAGTAANTAATGCCTTACCCATTTTTTGGTGAACTGACCCTTTA
>LH00444:71:2257TNLT4:6:1391:31783:21704/2
GATGAAAATGCAGTTGAGATATCAGAGGTGTAGTGTAGCTGTCTGTGGAGAGTACAGAAAGTGCTCAAAAAATCTTCTTTTTTTTCAGAAGATGAACGAAAGCGAGGGAGTAATTAATGCCTTACCCATTTTTTGGTGAACTGACCCTTTN
>LH00444:71:2257TNLT4:7:1307:30893:5539/2
GATGAAAATGCNGTTGAGATATCAGAGGTGTAGTGTAGCTGTCTGTGGAGAGTACAGAAAGTGCTCAAAAAATCTTCTTTTTTTTCAGAAGATGAACGAAAGCGAGGGAGTAATTAATGCCTTACCCATTTTTTGGTGAACTGACCCTTTA
>LH00444:71:2257TNLT4:7:1331:49513:19224/2
GATGAAAATGCAGTTGAGATATCAGAGGTGTAGTGTAGCTGTCTGTGGAGAGTACAGAAAGTGCTCAAAAAATCTTCTTTTTTTTCAGAAGATGAACGAAAGCGAGGGAGTAATTAATGCCTTACCCATTTTTTGGTGAACTGACCCTTTA
>LH00444:71:2257TNLT4:7:1206:23650:29534/1
AACAAAACAACACAATAGCTTTCAGCACAGCATTGAGCCACACAGACCTCGCTGCAGTGTCCTAAAACAATGGCTTAGCTTGAAAGCATCGGTCTTGTAATCCGAAGATCGAGGGTTAAATCCCCTCCTAGCAGA
>LH00444:71:2257TNLT4:7:1206:23650:29534/2
GCTAGGAGGGGATTTAACCCTCGATCTTCGGATTACAAGACCGATGCTTTCAAGCTAAGCCATTGTTTTAGGACACTGCAGCGAGGTCTGTGTGGCTCAATGCTGTGCTGAAAGCTATTGTGTTGTTTTGTTCCTGCAGCATGGCTCCAGA

~/seqtk/seqtk seq -a -q 10 -n N plate-11-GGAGCCATGCTGCAGG.fastq >  plate-11-GGAGCCATGCTGCAGG-q10.fasta
