---
title: "105-thalassinus"
output: html_document
date: "2024-04-11"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(ggpubr)
library(ggrepel)
library(phytools)
library(snpR)
library(ggtree)
library(ape)
library(phangorn)
library(tanggle)
```

```{r}
mt<-read_csv("meta/04152023.csv") %>% mutate(Path=paste0("data/align/",Ind,".sort.flt.bam"))  %>% mutate(Path2=ifelse(Dedup > 2e6, paste0("data/downsample/",Ind,".reduced.bam"),
                     Path)) %>%
  filter(Species%in% c("bicolor","thalassinus")) %>% filter(!`Tributary/Collection Location` %in% c("Bitner Ranch","Rock Creek","Horse Canyon","Thousand Creek Gorge","Cowhead Lake")) %>%
  mutate(Taxon=Species)

mt2<-mt %>% filter(Dedup > 2e5)
mt2 %>% group_by(Species, Subspecies, Vicinity, `Tributary/Collection Location`) %>% summarize(Count=n(), medianDedup=median(Dedup), meanDedup=mean(Dedup))

mt2 %>% group_by(Species, Subspecies, Vicinity, `Tributary/Collection Location`) %>% summarize(Count=n(), medianDedup=median(Dedup), meanDedup=mean(Dedup)) %>% write_csv("outputs/105/thal-meta.csv")
mt2 %>% select(Path2) %>% write_csv(col_names = FALSE, "bamlists/thal2.bamlist")
mt2$Vicinity<-factor(mt2$Vicinity, levels=c("Sheldon Wildlife Refuge","Harney County","Modoc County", "Warner Spring","Goose Lake","Rising River Preserve","Pit River","Valley Falls"))

mt2 %>% write_csv("meta/thalassinus.csv")
mt2 %>% select(SampleName) %>% write_tsv("meta/thalassinus-names.tsv", col_names = FALSE)
```


Cowhead Lake is something else entirely

100 inds

```{sh, eval=FALSE}
srun -p high -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/thal2.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-minInd 90 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.01 \
-out outputs/105/mt2-ibs-90 >outputs/105/mt2-ibs-90.out 2> outputs/105/mt2-ibs-90.err &

#-minMaf .01 not set
srun -p high -t 04:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/105/mt2-ibs-90.beagle.gz -K 2 -o outputs/105/mt2-K2 &
srun -p high -t 05:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/105/mt2-ibs-90.beagle.gz -K 3 -o outputs/105/mt2-K3 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/105/mt2-ibs-90.beagle.gz -K 4 -o outputs/105/mt2-K4 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/105/mt2-ibs-90.beagle.gz -K 5 -o outputs/105/mt2-K5 &
srun -p high -t 06:00:00 --mem=32G --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/105/mt2-ibs-90.beagle.gz -K 6 -o outputs/105/mt2-K6 &

```

151883 GLs


```{r}
m <- as.matrix(read.table("outputs/105/mt2-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(mt2)
bicolor<-covs %>% filter(Taxon=="bicolor") %>% group_by(Taxon) %>% summarize(x=mean(V1), y=mean(V2))
rr<-covs %>% filter(Taxon=="thalassinus") %>% group_by(Vicinity) %>% summarize(x=mean(V1), y=mean(V2))

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, shape=Taxon, fill=Vicinity), cex=3, alpha=0.75) +
  geom_text_repel(data=bicolor, aes(x=x, y=y), label="S. bicolor", max.overlaps = Inf) +
  geom_text_repel(data=rr, aes(x=x, y=y, label=Vicinity), max.overlaps = Inf) +

  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  scale_shape_manual(values=c(23,24,21)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("A") +
  theme(plot.title = element_text(hjust=0, size=16))

pc12
```

```{r}

bicolor13<-covs %>% filter(Taxon=="bicolor") %>% group_by(Taxon) %>% summarize(x=mean(V1), y=mean(V3))
rr13<-covs %>% filter(Taxon=="thalassinus") %>% group_by(Vicinity) %>% summarize(x=mean(V1), y=mean(V3))

pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, shape=Taxon, fill=Vicinity), cex=3, alpha=0.75) +
  geom_text_repel(data=bicolor13, aes(x=x, y=y), label="S. bicolor", max.overlaps = Inf) +
  geom_text_repel(data=rr13, aes(x=x, y=y, label=Vicinity), max.overlaps = Inf) +

  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  scale_shape_manual(values=c(23,24,21)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="turbo") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("B") +
  theme(plot.title = element_text(hjust=0, size=16))


pc13 
```

Scree plot

```{r}
vardf<-var %>% as_tibble() %>% rename(`Percent Variance`=value) %>% head() %>% mutate(PC=1:n())

scree<-ggplot(vardf, aes(x=PC, y=`Percent Variance`)) +
  geom_line() +
  geom_point() +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks = seq(1:6)) +
  theme(axis.text = element_text(size=10)) +
  theme(axis.title = element_text(size=14)) +
  ggtitle("C") +
  theme(plot.title = element_text(hjust=0, size=16))


scree
```

```{r}
ggarrange(pc12 + theme(legend.position = 'none'), 
          pc13, scree, ncol = 2, nrow=2, widths = c(1,1.4))

ggsave("outputs/105/thal-pcs.pdf", width = 11, height = 8.5)
```

## K plots


```{r, echo=FALSE}
PlotAdmix<- function(file, meta) {

q<-read_delim(file, delim=" ", col_names=FALSE)
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Last col empty
q<-q[1:length(colnames(q))-1]

#Bind met and arrange 
df<-bind_cols(q, meta) %>% mutate(Region=Vicinity) %>% arrange(Region) %>% mutate(Index=1:n())
#df$Region<-factor(df$Region, levels=unique(df$Region))

rdf<-df %>% dplyr::select(Region, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 3:(3+length(colnames(q))-1))

#Make names for structure-like plot
labels<-rdf %>% group_by(Region) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Region,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  ylab("Q\n") +
  theme(legend.position = "") +
  theme(axis.text = element_text(size=10, face="bold")) +
  theme(axis.title.y=element_text(size=14, face="bold")) +
  scale_x_continuous(breaks=labels$Position, labels=labels$Region) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  #theme(axis.text.x=element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  theme(axis.ticks.x = element_blank())

}
```


```{r}
PlotAdmix("outputs/105/mt2-K2.qopt", mt2)
```

```{r}
PlotAdmix("outputs/105/mt2-K3.qopt", mt2)
```


```{r}
PlotAdmix("outputs/105/mt2-K4.qopt", mt2)
```

```{r}
PlotAdmix("outputs/105/mt2-K5.qopt", mt2)

ggsave("outputs/105/Kplot-5.pdf")
```

```{r}
PlotAdmix("outputs/105/mt2-K6.qopt", mt2)
```

## Calls

We can place Rising River Preserve phylogenetically and get some metrics with calls (Fst, Pi). Hmm..

```{sh, eval=FALSE}
srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 90 -bam bamlists/thal2.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-out outputs/105/snps-100-01  \
-minMaf 0.01 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/105/snps-100-01.out 2> outputs/105/snps-100-01.err &

#82905 sites

#convert to .vcf
plink --tped snps-100-01.tped --tfam snps-100-01.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz

#prune (local)
bcftools reheader --samples meta/thalassinus-names.tsv outputs/105/plink.vcf.gz | bcftools +fill-tags | bcftools view -q 0.01:minor | bcftools +prune -m 0.20 -w 10000 > outputs/105/100-01.vcf

bcftools reheader --samples meta/thalassinus-names.tsv outputs/105/plink.vcf.gz | bcftools +fill-tags | bcftools view -q 0.01:minor | bcftools +prune -m 0.20 -w 10000 > outputs/105/100-01.vcf

conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i 100-01.vcf;
conda deactivate;

iqtree2 -s 100-01.min4.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo
iqtree2 -s 100-01.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 1000 -nm 2000 -bcor 0.9 --redo

#prune on farm
module load bcftools
bcftools reheader --samples meta/thalassinus-names.tsv outputs/105/plink.vcf.gz | bcftools +fill-tags -- -t MAF | bcftools view -q 0.01:minor |  bcftools +prune -m 0.3 -w 10000bp -n 1 -N maxAF > outputs/105/pruned-farm.vcf

bcftools reheader --samples meta/thalassinus-names.tsv outputs/105/plink.vcf.gz | bcftools +fill-tags -- -t MAF | bcftools view -q 0.10:minor |  bcftools +prune -m 0.3 -w 10000bp -n 1 -N maxAF > outputs/105/pruned10-farm.vcf

conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i pruned-farm.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i pruned10-farm.vcf;

conda deactivate;

iqtree2 -s pruned-farm.min4.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo
iqtree2 -s pruned-farm.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 1000 -nm 2000 -bcor 0.9 --redo

iqtree2 -s pruned10-farm.min4.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo
iqtree2 -s pruned10-farm.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 1000 -nm 2000 -bcor 0.9 --redo

```

1078 variable sites accoding to iqtree with local prune
4797 variable sites according to iqutree with prune on farm
3190 with MAF .10

```{r}
bis<-mt2 %>% filter(Species=="bicolor")
t<-read.tree("outputs/105/pruned-farm.min4.phy.varsites.phy.contree")
t<-root(t, bis$SampleName)
#t<-as.polytomy(t, feature='node.label', fun=function(x) as.numeric(x) < 50)

u<-ggtree(t) %<+% (mt2 %>% relocate(SampleName)) 

u +  geom_tippoint(aes(fill=Vicinity), pch=21) +
  scale_fill_viridis_d(option="H")

ggsave("outputs/105/tree.pdf")
```

```{r}
dat<-read.dna("outputs/105/pruned-farm.min4.phy")
write.nexus.data(dat, file="outputs/105/pruned-farm.nex")

dat<-read.dna("outputs/105/pruned10-farm.min4.phy")
write.nexus.data(dat, file="outputs/105/pruned10-farm.nex")

```

```{r}
net<-read.nexus.networx("outputs/105/farm.network")
net<-read.nexus.networx("outputs/105/farm-pruned10.network")

```

```{r}
g<-ggsplitnet(net, color="grey50")  

g$data<-left_join(g$data, mt2, by=c("label"="SampleName")) 

```


```{r}
n<-g + 
  geom_tippoint(aes(fill=Vicinity), pch=21, cex=5, alpha=1.0) +
  scale_fill_viridis_d(option="H") 


n
ggsave("outputs/105/thal-network.pdf", width=8.5, height=11)

```
## Lower expectations
Sample sizes

```{r}
mt %>% filter(Dedup > 1.5e5) %>% group_by(Species, Subspecies, Vicinity, `Tributary/Collection Location`) %>% summarize(Count=n(), medianDedup=median(Dedup), meanDedup=mean(Dedup))
```