---
title: "106-cytb-species-delim"
output: html_document
date: "2024-05-01"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(tidyverse)
library(ape)
library(pegas)
library(ggtree)
library(phytools)
library(phylotools)
library(viridis)
```

Redoing cytb analysis with outgroups. Including Chrosomus and Orthodon
`>OR002152.1:14388-15526 Chrosomus erythrogaster mitochondrion, complete genome
ATGGCAAGCCTACGAAAAACCCACCCACTAATGAAAATCGCTAATGACGCATTAGTTGACCTCCCGACACCATCTAACAT
TTCCGCACTCTGAAACTTTGGATCCCTCCTAGGATTATGTTTAATTACCCAAATCCTAACAGGATTATTTTTAGCCATAC
ACTATACCTCTGATATCTCAACTGCATTTTCATCAGTTACTCACATCTGCCGGGACGTTAACTACGGCTGACTTATTCGA
AACCTACATGCCAACGGTGCATCATTCTTCTTCATCTGCATTTATATGCACATCGCCCGAGGCCTGTACTACGGGTCATA
CCTCTATAAAGAAACCTGAAACATTGGCGTAGTACTACTTCTATTAGTTATAATAACGGCCTTCGTCGGTTACGTCCTAC
CATGAGGCCAAATGTCTTTTTGAGGCGCCACAGTAATTACAAATTTACTCTCAGCAGTACCCTACATAGGTGACACCCTC
GTGCAATGAATCTGGGGAGGCTTCTCAGTAGACAACGCAACACTCACACGATTTTTCGCCTTCCACTTCCTACTACCCTT
CGTAATTGCCGGCGCAACCGTCCTCCACCTGCTATTTCTCCACGAAACCGGCTCGAACAACCCGGCCGGGCTAAACTCTG
ATGCAGATAAAATCTCTTTTCATCCATACTTTTCATACAAAGACCTTCTTGGTTTCGTACTAATATTATTTGCCCTCACG
TCATTAGCATTATTTTCTCCCAACTTATTGGGTGACCCAGAGAATTTTACCCCTGCAAATCCCCTAGTAACACCCCCACA
TATTCAACCTGAATGATATTTCTTATTTGCCTATGCCATTCTCCGTTCCATCCCAAACAAACTAGGTGGGGTCCTTGCAC
TACTATTCAGCATTCTTGTGCTACTAGTAGTACCAATTTTACACACCTCGAAACAACGAGGACTTACTTTCCGCCCTATG
ACCCAATTCCTATTCTGAACCCTAGTAGCAGATATACTTATCTTAACATGAATTGGCGGTATACCTGTAGAACACCCATA
TATTATTATTGGCCAAGTAGCATCAATTCTATATTTTGCACTTTTCCTCATTCTTGTCCCACTAGCGGGATGAGTAGAAA
ATAAGGCATTAAAATGAGC`

`>AF452073.1 Orthodon microlepidotus cytochrome b (cytb) gene, partial cds; mitochondrial gene for mitochondrial product
ATGGCAAGCCTACGAAAAACCCACCCACTAATAAAAATCGCTAACGATGCGCTAGTCGACCTTCCAACAC
CGTCCAATATTTCAGCACTATGAAATTTTGGCTCCCTCCTAGGATTATGTTTAATTACCCAAATTCTAAC
AGGGTTATTTTTAGCCATACACTACACCTCCGATATCTCGACCGCATTTTCATCAGTGACTCACATCTGC
CGAGACGTTAACTATGGCTGACTTATCCGAAGCCTACATGCCAACGGAGCATCATTTTTCTTCATCTGCA
TTTATATACACATTGCCCGGGGCCTATATTATGGGTCATACCTTCATAAAGAAACCTGAAACATTGGCGT
AGTCCTACTTCTTTTAGTTATAATGACCGCATTCGTTGGCTATGTACTTCCGTGAGGCCAAATATCTTTT
TGAGGCGCCACAGTAATTACAAATCTGCTATCCGCAGTCCCTTACATAGGAGACACCCTTGTTCAGTGAA
TCTGAGGAGGCTTCTCAGTAGACAACGCAACACTGACACGATTCTTCGCCTTCCACTTCCTACTACCGTT
TGTCATCGCCGGCGCGACCATCCTACACTTACTGTTCTTACACGAAACCGGGTCAAACAACCCGGCCGGA
CTGAACTCTGATGCGGATAAAATTTCTTTTCATCCATACTTCTCATACAAAGACCTTCTTGGCTTCGTAC
TAATATTACTAGCCCTCACTTCACTAGCATTGTTTTCACCCAACCTGCTAGGTGACTCAGAAAATTTTAT
TCCGGCAAACCCCTTAGTCACCCCTCCACATATTCAGCCTGAATGATATTTCCTATTTGCCTACGCCATT
CTACGATCTATCCCTAACAAACTAGGAGGGGTCCTCGCACTACTATTTAGCATTCTAGTGCTATTGGTTG
TGCCAATCTTACATACCTCAAAACAACGAGGATTAACTTTCCGTCCCATGACCCAATTTCTATTCTGAAC
CCTAGTAGCAGATATACCCATTTTGACATGAATCGGCGGCATACCCGTAGAACACCCTTATATTATTATT
GGCCAAGCCGCATCAATCCTATACTTCGCACTATTTCTTATTCTAGTCCCACTAGCTGGGTGAGTAGAAA
ATAAAGCGTTGAAATGAGCT`


Get the ingroup and do some species delimitation, using mafft v 7.  
Dropped some of the Ptychocheilus 

`mafft accessions-with-extras.fas > accessions-with-extras.afas`
`iqtree2 -T AUTO -m GTR+G -s accessions-with-extras.afas -bb 10000 -redo`
`iqtree2 -T AUTO -m MFP -s accessions-with-extras.afas -bb 10000 -redo`

`mafft reduced-ogs.fas > reduced-ogs.afas`
`iqtree2 -T AUTO -m GTR+G -s reduced-ogs.afas -bb 10000 -redo`
Removed AF237737, duplicate sequence with AF237737.1
`iqtree2 -T AUTO -m MFP -s reduced-ogs.afas -bb 10000 -redo`
```{r}
meta<-read_tsv("outputs/106/annotations-with-extas.tsv")

meta<-meta %>% separate(Fish, into=c("Genus","Species", "subspecies","ID"), remove = FALSE)

bicolor<-meta %>% filter(Species=="bicolor")
write_csv(bicolor,"meta/cytb-meta.csv")
ogs<-meta %>% filter(Species!="bicolor")
```
```{r}
bicolor<-read_csv("meta/cytb-meta-edited.csv")
bicolor$NewLocation<-gsub("Nevada","NV",bicolor$NewLocation)
bicolor$NewLocation<-gsub("California","CA",bicolor$NewLocation)
bicolor$NewLocation<-gsub("Oregon","OR",bicolor$NewLocation)
bicolor$NewLocation<-gsub("County","Co.",bicolor$NewLocation)

bicolor %>% select(-Location)
bicolor$Location<-bicolor$NewLocation
```

https://bioinfo.mnhn.fr/abi/public/asap/asapweb.html

Threshold dist P2 0.011930 supports six species
Threshold dist P3 0.016382 supports five species
Threshold Dist P1 (subset 52, 0.000878)
```{r}
species<-read_csv("cytb-species-delim/extra-mohave.afas.Partition_3.csv",
                  col_names = c("Tip","Subset"))
m2<-left_join(meta, species)
m2$Subset<-as.factor(m2$Subset)
```

```{r}
#tree<-midpoint.root(read.tree("outputs/106/accessions-with-extras.afas.contree"))
tree<-read.tree("outputs/106/accessions-with-extras.afas.contree")
tree<-read.tree("outputs/106/reduced-ogs.afas.contree")
tree<-midpoint.root(tree)
u<-as.polytomy(tree, feature='node.label', fun=function(x) as.numeric(x) < 50)

v<-ggtree(u) %<+% m2 

v 
```

```{r}
tips<-v$data %>% filter(isTip==TRUE) %>% left_join(bicolor, by=c("label"="Tip"))
tips
```
##write_csv(tips, file="outputs/106/tips.csv")

```{r}
t<-v
d <- t$data
d <- d[!d$isTip,]
d$label <- as.numeric(d$label)
d <- d[d$label >= 90,]

e <- t$data
e <- e[!e$isTip,]
e$label <- as.numeric(e$label)
e <- e[e$label < 90,]
e <- e[e$label >= 75,]
```

```{r}
cols<-viridis(10, option="H")


```

```{r}
v    %<+% bicolor +
  geom_point(data=e,  fill="white", cex=2, alpha=1, pch=21) +
  geom_point(data=d,  fill="grey50", cex=2, alpha=1, pch=21) +
  geom_tiplab(data=tips, aes(x=.15, y=y, label=NewLocation, color=Subset), size=2) +
  geom_tiplab(size=2) +
  xlim(0,.25) +
  scale_colour_viridis_d(option="H") +
  theme(legend.position = "none") +
  scale_color_manual(values=c(cols[9],cols[10],cols[7],cols[6],cols[5],cols[4]), na.value = NA) 


ggsave("outputs/106/cyt-b-tree-with-delimitations.pdf", width=8.5, height=11)
```
create my colors, need 6 based on viridis H for 10





```{r}
v + geom_point(aes(color=Subset), x=0.14, pch=15) +
  geom_point(data=e,  fill="white", cex=2, alpha=1, pch=21) +
  geom_point(data=d,  fill="grey50", cex=2, alpha=1, pch=21) +
  geom_tiplab(data=v$data %>% filter(Species != "bicolor"), aes(x=x+0.001,label=Fish), size=2) +
  theme(legend.position = "none") +
  scale_color_manual(values=c(cols[9],cols[10],cols[7],cols[6],cols[5],cols[4]), na.value = NA) +
  xlim(0, 0.25)

ggsave("outputs/106/cyt-b-tree-with-delimitations-no-tips.pdf", width=8.5, height=8)
ggsave("outputs/106/cyt-b-tree-with-delimitations-no-tips.jpeg", width=8.5, height=8)

```


We can use:

Siphateles mohavensis
S. isolatus
S. newarkensis (newarkensis newarkensis, newarkensis euchila)
See page 40 of harris 2000
S. thalassinus
S. bicolor (OS-16665) -(bicolor, bicolor, bicolor subsp. silver lake valley,  bicolor colubmianus, bicolor eurysoma).
S. obesus (what we have genomic data for)
  Includes "oregonensis" from Abert Lake (XL Springs) OS 16774, "snyderi" from Owens Basin, Owens River (OS15747), "pectinifer" BB48,49 from Lahontan Basin


## Some new seqs

```{r}
#tree<-midpoint.root(read.tree("outputs/106/accessions-with-extras.afas.contree"))
tree<-read.tree("/Users/mac/Dropbox/ucdavis/tui-chub/TuiChubSanger_FiveSuccessfulSamples_FASTA/adding-to-existing/comb-trim.afas.contree")
tree<-midpoint.root(tree)
u<-as.polytomy(tree, feature='node.label', fun=function(x) as.numeric(x) < 50)

v<-ggtree(u) %<+% m2 

v 
```

```{r}
v    %<+% bicolor +
  geom_tiplab(size=2) +
  geom_tiplab(aes(x=.15, y=y, label=NewLocation), size=2) +

  xlim(0,.25) 

ggsave("outputs/106/trial-new-cytb-649chars.pdf", width=8.5, height=11)

```
# RAG1

 perl ~/Dropbox/bin/shortenNamesMac.pl revised-western-clade.fas > revised-western-clade-renamed.fas
mafft revised-western-clade-renamed.fas > revised-western-clade.afas
iqtree2 -T AUTO -m GTR+G -s revised-western-clade.afas -bb 10000 -redo