---
title: "107-snp-phylogeny"
output: html_document
date: "2024-05-01"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(poppr)
library(vcfR)
library(phytools)
library(phangorn)
library(ape)
library(ggtree)
library(tanggle)
```

Let's design a phylogeny with some downsampled tips.

Outgroups
G. orcuttii
S. alvordensis
P. macrolepidotus

Ingroups
isolatus
newarkensis (NN4, NV12 combined)
thalassinus vaccaceps (separate)
thalassinus  (combined locs)
bicolor columbianus Kueney County
bicolor eurysomas (Combined locs)
mohavensis Camp Cady, Tui Slough 
for toikona let's use Mule Spring / Cottonwood Pond combined
for snyderi let's use Sotcher Lake, NE Pond, SW Pond, SE Pond combined
for obesus 
Fish Lake Valley Lida Pond,
Fish Lake Valley McNett Ranch
Little Fish Lake Valley
Hot Creek Valley/Railroad Valley

Combined Lahontan lineage
Walker Lake/Rose Creek 
Eagle Lake 
Carson Desert, Little Soda Lake
Upper Humboldt

```{r}
pm<-read_csv("meta/05012024.csv") %>% mutate(Path=paste0("data/align/",Ind,".sort.flt.bam")) %>% 
   mutate(Path2=ifelse(Dedup > 2e6, paste0("data/downsample/",Ind,".reduced.bam"),
                     Path)) %>% filter(Dedup>1e6) %>%  mutate(Sample=paste0(1:n(),"_",1))
nrow(pm)
```


```{r}
pm %>% group_by(Species) %>% summarize(Count=n())
```

# Rooted Phylogeny

## Outgroups

```{r}
gila<-pm %>% filter(Sample %in% c("634_1","617_1"))
alvo<-pm %>% filter(Species =="alvordensis")
pogo<-pm %>% filter(Species =="P. macrolepidotus")
ogsdf<-bind_rows(gila, alvo, pogo)
nodownsample<-pm %>% filter(Species %in% c("bicolor","thalassinus", "isolatus"))
#get pops newarkensis, mohavensis
# NN4 has 20 inds, can consider adding in an other location, NV12?
new<-m1 %>% filter(`Tributary/Collection Location` %in% c("NN4","NV12","Tui Slough", "Camp Cady")) %>% filter(Sample != "237_1")
```

## Ingroup

```{r}
toikona<-pm %>% filter(Subspecies=="toikona")
snyderi<-pm %>% filter(Subspecies=="snyderi") %>% filter(`Tributary/Collection Location` %in% c("Sotcher Lake","Sotcher Lake Outlet, Reds Meadow Creek",
                                                                                                "NE Pond (White Mountain Research Center)","SE Pond (White Mountain Research Center)","SW Pond (White Mountain Research Center)"))
```

```{r}
fishlake<-pm %>% filter(Species=="obesus") %>% filter(`Tributary/Collection Location` %in% c("McNett Ranch","Lida Pond","Little Fish Lake"))
railroad<-pm %>% filter(Species=="obesus") %>% filter(Vicinity %in% c("Railroad Valley", "Hot Creek Valley"))
lahontan<-pm %>% filter(Species=="obesus") %>% filter(`Tributary/Collection Location` %in% c("Upper Humboldt River","Rose Creek Reservoir","Little Soda Lake",
                                                                                             "Eagle Lake"))
obesus<-bind_rows(toikona, snyderi, fishlake, railroad, lahontan)
```

Combined Lahontan lineage
Walker Lake/Rose Creek 
Eagle Lake 
Upper Humboldt
Little Soda Lake

```{r}
phylo<-bind_rows(gila, alvo, pogo, nodownsample,new,toikona, snyderi, fishlake, railroad, lahontan) %>% mutate(Sample=paste0(1:n(),"_",1))
write_csv(phylo %>% select(Sample), file="outputs/107/438.names", col_names=FALSE)
write_tsv(phylo %>% select(Path2), file="bamlists/438.bamlist", col_names = FALSE)
write_csv(phylo, file="meta/meta-438.csv")
```

Call snps at 95% threshold with a minMaf of 0.01.   

```{sh, eval=FALSE}
srun -t 36:00:00 -p bigmemh --mem=128G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 416 -bam bamlists/438.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-out outputs/107/snps-438-01  \
-minMaf 0.01 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/107/snps-438-01.out 2> outputs/107/snps-438-01.err &

#convert to .vcf
plink --tped snps-438-01.tped --tfam snps-438-01.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz

#prune
bcftools  view -S outputs/107/438.names outputs/107/plink.vcf.gz | bcftools +fill-tags | bcftools view -q 0.01:minor | bcftools +prune -m 0.20 -w 10000 > outputs/107/438-01.vcf

#trial phylogeny

conda activate py2
~/github/mccloud-rrt/vcf2phylip.py -i 438-01.vcf

conda deactivate

iqtree2 -s 438-01.min4.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo
iqtree2 -s 438-01.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 1000 -nm 2000 -bcor 0.9 --redo

```

438 18845.   



```{r}
t<-read.tree("outputs/107/438-01.min4.phy.varsites.phy.contree")
ogs<-phylo %>% filter(GVL_Code %in% ogsdf$GVL_Code) %>% select(Sample)

#t<-midpoint_root(t) 
t<-root(t, ogs$Sample)
t<-as.polytomy(t, feature='node.label', fun=function(x) as.numeric(x) < 50)

ggtree(t) %<+% (phylo %>% relocate(Sample)) +
  
  geom_tippoint(aes(fill=Species), pch=21) +
  geom_tiplab(aes(label=paste0(Species," ",Vicinity," ",`Tributary/Collection Location`), x=.2), size=.75) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  geom_treescale() +
 # geom_nodelab() +
  xlim(0,.3) 

ggsave("outputs/107/438-ml-tree.pdf", height=20, width=8.5)
```


## Implicit Network

```{r}
dat<-read.dna(file="outputs/107/438-01.min4.phy.varsites.phy")
write.nexus.data(dat, file="outputs/107/438-01.nex")
```

```{r}
net<-read.nexus.networx("outputs/107/438-01.network")
```

```{r}
g<-ggsplitnet(net)  
g$data<-left_join(g$data, phylo, by=c("label"="Sample")) 

n<-g + 
  geom_tippoint(aes(fill=Species), cex=5, pch=21, alpha=1.0) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22)))

n

ggsave("outputs/107/438-network.pdf", width=12, height=10)
```

```{r}
g2<-ggsplitnet(net)  
phylo2<-phylo
phylo2$Species<-factor(phylo2$Species, levels=c("G. orcuttii","P. macrolepidotus","alvordensis","thalassinus","bicolor","isolatus","newarkensis","mohavensis","obesus"))
g2$data<-left_join(g2$data, phylo2, by=c("label"="Sample")) 

n2<-g2 + 
  geom_tippoint(aes(shape=Species, fill=Subspecies), cex=5, alpha=1.0) +
  scale_fill_viridis_d(option="magma", na.value="grey50") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  scale_shape_manual(values=c(24,24,24,24,22,22,22,22,21))

n2

ggsave("outputs/107/438-obesus-subs-network.pdf", width=12, height=10)

```

## Explicit Network



