---
title: "107-snp-phylogeny"
output: html_document
date: "2024-05-01"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(ggrepel)
library(poppr)
library(vcfR)
library(phytools)
library(phangorn)
library(ape)
library(ggtree)
library(tanggle)
```

Let's design a phylogeny with some downsampled tips.

Outgroups
G. orcuttii
S. alvordensis
P. macrolepidotus

Ingroups
isolatus
newarkensis (NN4, NV12 combined)
thalassinus vaccaceps (separate)
thalassinus  (combined locs)
bicolor columbianus Kueney County
bicolor eurysomas (Combined locs)
mohavensis Camp Cady, Tui Slough 
for toikona let's use Mule Spring / Cottonwood Pond combined
for snyderi let's use Sotcher Lake, NE Pond, SW Pond, SE Pond combined
for obesus 
Fish Lake Valley Lida Pond,
Fish Lake Valley McNett Ranch
Little Fish Lake Valley
Hot Creek Valley/Railroad Valley

Combined Lahontan lineage
Walker Lake/Rose Creek 
Eagle Lake 
Carson Desert, Little Soda Lake
Upper Humboldt

```{r}
pm<-read_csv("meta/05012024.csv") %>% mutate(Path=paste0("data/align/",Ind,".sort.flt.bam")) %>% 
   mutate(Path2=ifelse(Dedup > 2e6, paste0("data/downsample/",Ind,".reduced.bam"),
                     Path)) %>% filter(Dedup>1e6) %>%  mutate(Sample=paste0(1:n(),"_",1))
nrow(pm)
```


```{r}
pm %>% group_by(Species) %>% summarize(Count=n())
```

# Rooted Phylogeny

## Outgroups

```{r}
gila<-pm %>% filter(Sample %in% c("634_1","617_1"))
alvo<-pm %>% filter(Species =="alvordensis")
pogo<-pm %>% filter(Species =="P. macrolepidotus")
ogsdf<-bind_rows(gila, alvo, pogo)
nodownsample<-pm %>% filter(Species %in% c("bicolor","thalassinus", "isolatus"))
#get pops newarkensis, mohavensis
# NN4 has 20 inds, can consider adding in an other location, NV12?
new<-m1 %>% filter(`Tributary/Collection Location` %in% c("NN4","NV12","Tui Slough", "Camp Cady")) %>% filter(Sample != "237_1")
```

## Ingroup

```{r}
toikona<-pm %>% filter(Subspecies=="toikona")
snyderi<-pm %>% filter(Subspecies=="snyderi") %>% filter(`Tributary/Collection Location` %in% c("Sotcher Lake","Sotcher Lake Outlet, Reds Meadow Creek",
                                                                                                "NE Pond (White Mountain Research Center)","SE Pond (White Mountain Research Center)","SW Pond (White Mountain Research Center)"))
```

```{r}
fishlake<-pm %>% filter(Species=="obesus") %>% filter(`Tributary/Collection Location` %in% c("McNett Ranch","Lida Pond","Little Fish Lake"))
railroad<-pm %>% filter(Species=="obesus") %>% filter(Vicinity %in% c("Railroad Valley", "Hot Creek Valley"))
lahontan<-pm %>% filter(Species=="obesus") %>% filter(`Tributary/Collection Location` %in% c("Upper Humboldt River","Rose Creek Reservoir","Little Soda Lake",
                                                                                             "Eagle Lake"))
obesus<-bind_rows(toikona, snyderi, fishlake, railroad, lahontan)
```

Combined Lahontan lineage
Walker Lake/Rose Creek 
Eagle Lake 
Upper Humboldt
Little Soda Lake

```{r}
phylo<-bind_rows(gila, alvo, pogo, nodownsample,new,toikona, snyderi, fishlake, railroad, lahontan) %>% mutate(Sample2=Sample) %>% mutate(Sample=paste0(1:n(),"_",1))
write_csv(phylo %>% select(Sample), file="outputs/107/438.names", col_names=FALSE)
write_csv(phylo %>% select(Sample), file="meta/438.names", col_names=FALSE)

write_tsv(phylo %>% select(Path2), file="bamlists/438.bamlist", col_names = FALSE)
write_csv(phylo, file="meta/meta-438.csv")

phylo %>% group_by(`Tributary/Collection Location`) %>% mutate(SVDQ=paste0(Sample,"T")) %>% select(SVDQ) %>%  summarize(Inds=paste0(SVDQ, collapse="\t"))  %>% write_tsv("outputs/107/block.tsv")

```

Call snps at 95% threshold with a minMaf of 0.01.   

```{sh, eval=FALSE}
srun -t 36:00:00 -p bigmemh --mem=128G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 416 -bam bamlists/438.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-out outputs/107/snps-438-01  \
-minMaf 0.01 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/107/snps-438-01.out 2> outputs/107/snps-438-01.err &

#convert to .vcf
plink --tped snps-438-01.tped --tfam snps-438-01.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz

#prune (local)
bcftools  view -S outputs/107/438.names outputs/107/plink.vcf.gz | bcftools +fill-tags | bcftools view -q 0.01:minor | bcftools +prune -m 0.20 -w 10000 > outputs/107/438-01.vcf

bcftools  view -S outputs/107/438.names outputs/107/plink.vcf.gz | bcftools +fill-tags | bcftools view -q 0.01:minor | bcftools +prune -m 0.20 -w 10000 > outputs/107/438-01-redo.vcf

#manipulating here:
bcftools  view -S outputs/107/438.names outputs/107/plink.vcf.gz | bcftools +fill-tags | bcftools view -q 0.10:minor | bcftools +prune -m 0.20 -w 10000 > outputs/107/438-10.vcf
bcftools  view -S outputs/107/438.names outputs/107/plink.vcf.gz | bcftools +fill-tags | bcftools view -q 0.05:minor | bcftools +prune -m 0.20 -w 10000 > outputs/107/438-05.vcf

bcftools  view -S outputs/107/438.names outputs/107/plink.vcf.gz | bcftools +fill-tags | bcftools view -q 0.01:minor | bcftools +prune -m 0.20 -w 10000bp > outputs/107/438-01-bp.vcf


#prune on farm
module load bcftools
bcftools reheader --samples meta/438.names outputs/107/plink.vcf.gz | bcftools +fill-tags -- -t MAF | bcftools view -q 0.01:minor |  bcftools +prune -m 0.2 -w 10000bp -n 1 -N maxAF > outputs/107/pruned-farm.vcf
bcftools reheader --samples meta/438.names outputs/107/plink.vcf.gz | bcftools +fill-tags -- -t MAF | bcftools view -q 0.1:minor |  bcftools +prune -m 0.2 -w 10000bp -n 1 -N maxAF > outputs/107/pruned-farm-10.vcf  

#trial phylogeny

conda activate py2
~/github/mccloud-rrt/vcf2phylip.py -i pruned-farm.vcf
~/github/mccloud-rrt/vcf2phylip.py -i pruned-farm-10.vcf

~/github/mccloud-rrt/vcf2phylip.py -i 438-01.vcf
~/github/mccloud-rrt/vcf2phylip.py -i 438-05.vcf
~/github/mccloud-rrt/vcf2phylip.py -i 438-10.vcf

conda deactivate

iqtree2 -s 438-01.min4.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo
iqtree2 -s 438-01.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 1000 -nm 2000 -bcor 0.9 --redo

iqtree2 -s 438-05.min4.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo
iqtree2 -s 438-05.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 1000 -nm 2000 -bcor 0.9 --redo
#Alignment has 438 sequences with 2649 columns, 2649 distinct patterns
#2497 parsimony-informative, 152 singleton sites, 0 constant sites


iqtree2 -s 438-10.min4.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo
iqtree2 -s 438-10.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 1000 -nm 2000 -bcor 0.9 --redo
#Alignment has 438 sequences with 1707 columns, 1707 distinct patterns
#1599 parsimony-informative, 108 singleton sites, 0 constant sites


iqtree2 -s pruned-farm.min4.phy -T AUTO -st DNA -m MFP+ASC -bb 10000 --redo
iqtree2 -s pruned-farm.min4.phy.varsites.phy -T AUTO -st DNA -m GTR+ASC -bb 1000 -nm 2000 -bcor 0.9 --redo
#Alignment has 438 sequences with 9329 columns, 9237 distinct patterns



```

438 18845.   



```{r}
t<-read.tree("outputs/107/438-05.min4.phy.varsites.phy.contree")
ogs<-phylo %>% filter(GVL_Code %in% ogsdf$GVL_Code) %>% select(Sample)
p<-phylo %>% filter(Species %in%  c("P. macrolepidotus","G. orcuttii"))
#t<-midpoint_root(t) 
t<-root(t, p$Sample)
t<-as.polytomy(t, feature='node.label', fun=function(x) as.numeric(x) < 50)

ggtree(t) %<+% (phylo %>% relocate(Sample)) +
  
  geom_tippoint(aes(fill=Species), pch=21) +
  geom_tiplab(aes(label=paste0(Species," ",Vicinity," ",`Tributary/Collection Location`), x=.35), size=.75) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  geom_treescale() +
 # geom_nodelab() +
  xlim(0,.5) 

ggsave("outputs/107/438-ml-tree.pdf", height=20, width=8.5)
```



## Implicit Network

```{r}
dat<-read.dna(file="outputs/107/438-01.min4.phy.varsites.phy")
write.nexus.data(dat, file="outputs/107/438-01.nex")

dat<-read.dna(file="outputs/107/pruned-farm.min4.phy.varsites.phy")
write.nexus.data(dat, file="outputs/107/pruned-farm.nex")

dat<-read.dna(file="outputs/107/438-05.min4.phy")
write.nexus.data(dat, file="outputs/107/438-05.nex")


dat<-read.dna(file="outputs/107/438-10.min4.phy")
write.nexus.data(dat, file="outputs/107/438-10.nex")

```

```{r}
net<-read.nexus.networx("outputs/107/438-01.network")
```

```{r}
nm<-read_csv("meta/meta-438-edited.csv")
nm$Species<-factor(nm$Species, levels = c("P. macrolepidotus","G. orcuttii","alvordensis","thalassinus","bicolor","newarkensis","isolatus","snyderi","mohavensis","obesus"))
nm<-nm %>% mutate(Taxon=Species)

nm2<-read_csv("meta/meta-438-edited-06272024.csv")
nm2$Species<-factor(nm2$Species, levels = c("P. macrolepidotus","G. orcuttii","alvordensis","thalassinus","bicolor","newarkensis","isolatus","snyderi","mohavensis","obesus"))
nm2<-nm2 %>% mutate(Taxon=Species)

nm2 %>% group_by(Taxon, Vicinity, `Tributary/Collection Location`) %>% summarize(MedianReadCount=round(median(Dedup),0), SampleSize=n())  %>% write_csv("meta/438-sample-table.csv")
```



replotting tree
```{r}

v<-ggtree(t) %<+% (nm %>% relocate(Sample)) 

  
t<-v
d <- t$data
d <- d[!d$isTip,]
d$label <- as.numeric(d$label)
d <- d[d$label >= 90,]

e <- t$data
e <- e[!e$isTip,]
e$label <- as.numeric(e$label)
e <- e[e$label < 90,]
e <- e[e$label >= 75,]
```


```{r}
  v+
  geom_point(data=e,  fill="white", cex=2, alpha=1, pch=21) +
  geom_point(data=d,  fill="grey50", cex=2, alpha=1, pch=21) +
  geom_tippoint(aes(fill=Taxon, shape=Taxon, x=x+0.001), cex=3) +
 # geom_tiplab(aes(label=paste0(Species," ",Vicinity," ",`Tributary/Collection Location`), x=.4), size=.75) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  geom_treescale() +
  scale_shape_manual(values=c(22,22,23,25,25,24,24,21,21,21)) +

 # geom_nodelab() +
  xlim(0,.4) +
  ylim(0, max(v$data$y))

ggsave("outputs/107/438-ml-tree-05.pdf", height=12, width=8.5)
```

Generating tree with tip labels

```{r}
  v+
  geom_point(data=e,  fill="white", cex=2, alpha=1, pch=21) +
  geom_point(data=d,  fill="grey50", cex=2, alpha=1, pch=21) +
 # geom_tippoint(aes(fill=Taxon, shape=Taxon, x=x+0.001), cex=3) +
  geom_tiplab(aes(label=paste0(Vicinity," ",`Tributary/Collection Location`), x=x+.006, color=Taxon), size=2) +
  scale_color_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  geom_treescale() +
  scale_shape_manual(values=c(22,22,23,25,25,24,24,21,21,21)) +

 # geom_nodelab() +
  xlim(0,.4) +
  ylim(0, max(v$data$y))

ggsave("outputs/107/438-ml-tree-05-tips.pdf", height=36, width=8.5)
```

```{r}
g<-ggsplitnet(net, color="grey50")  
g$data<-left_join(g$data, nm, by=c("label"="Sample")) 

labels<-g$data %>% filter(isTip==TRUE) %>% group_by(Species) %>% summarize(x=mean(x), y=mean(y)) %>% filter(Species !="obesus")
labels2<-g$data %>% filter(isTip==TRUE) %>% group_by(Species, `Tributary/Collection Location`) %>% summarize(x=mean(x), y=mean(y)) %>% filter(Species !="obesus")

olabels<-g$data %>% filter(isTip==TRUE) %>% group_by(Vicinity, `Tributary/Collection Location`, Species) %>% summarize(x=mean(x), y=mean(y)) %>% filter(Species =="obesus") %>% 
  filter(`Tributary/Collection Location`!="Flowing Well #7")

```

```{r}
n<-g + 
  geom_tippoint(aes(fill=Taxon, shape=Taxon), cex=5, alpha=1.0) +
  geom_text_repel(data=olabels, aes(x=x, y=y, label=`Tributary/Collection Location`)) +
  geom_text_repel(data=labels, aes(x=x, y=y, label= Species)) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  scale_shape_manual(values=c(22,22,23,25,25,24,24,21,21,21))


n

ggsave("outputs/107/438-network.pdf", width=10, height=6)
```


# SVDQuartets trees

maf05 and maf10

Running maf05 /github/fish-lake/outputs/107/svdq-05-nomin.txt for 100 boostraps in backgroun

```{r}
st<-read_tsv("meta/species-tree-labels")
st$Taxon<-factor(st$Taxon, levels = c("P. macrolepidotus","G. orcuttii","alvordensis","thalassinus","bicolor","newarkensis","isolatus","snyderi","mohavensis","obesus"))

st
```

```{r}
s<-read.nexus("outputs/107/svdq-boot-05-nomin.tre")
s<-consensus(s, p=0.5) # p can't be less than .5

#s<-read.nexus("outputs/107/svdq-boot-10.tre")
s<-root(s, c("P_macrolepidotus","G_orcuttii","E_alvordensis"))

#ggtree(s, branch.length = 'none') %<+% block +
 # geom_tippoint(aes(shape=Species, fill=Subspecies, size=Count)) + 
ggtree(s, branch.length = "none") %<+% st +  
  geom_tippoint(aes(fill=Taxon, shape=Taxon), cex=4) + 
  geom_tiplab(size=4, offset = 0.5) +
  xlim(0,20) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  scale_shape_manual(values=c(22,22,23,25,25,24,24,21,21,21)) 



ggsave("outputs/107/svdq-maf05.pdf", width=11, height = 8.5)
ggsave("outputs/107/svdq-maf05.jpeg", width=11, height = 8.5)

```

```{r}
s<-read.nexus("outputs/107/svdq-boot-10.tre")
s<-consensus(s, p=0.5) # p can't be less than .5

#s<-read.nexus("outputs/107/svdq-boot-10.tre")
s<-root(s, c("P_macrolepidotus"))

#ggtree(s, branch.length = 'none') %<+% block +
 # geom_tippoint(aes(shape=Species, fill=Subspecies, size=Count)) + 
ggtree(s, branch.length = "none") +  
  geom_nodelab() + 
  geom_tiplab(size=4, offset = 0.5) +
  xlim(0,20) +
  scale_fill_viridis_d(option="magma", na.value="grey50") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  scale_shape_manual(values=c(25,23,25,23,24,22,21,23,25))



ggsave("outputs/107/svdq-maf10.pdf", width=11, height = 8.5)

```


# Treemix


# Alternative maf of 0.05


```{r}
net<-read.nexus.networx("outputs/107/438-05.network")
```


```{r}
g<-ggsplitnet(net, color="grey50")  
g$data<-left_join(g$data, nm, by=c("label"="Sample")) 

labels<-g$data %>% filter(isTip==TRUE) %>% group_by(Species) %>% summarize(x=mean(x), y=mean(y)) %>% filter(Species !="obesus")
labels2<-g$data %>% filter(isTip==TRUE) %>% group_by(Species, `Tributary/Collection Location`) %>% summarize(x=mean(x), y=mean(y)) %>% filter(Species !="obesus")

olabels<-g$data %>% filter(isTip==TRUE) %>% group_by(Vicinity, `Tributary/Collection Location`, Species) %>% summarize(x=mean(x), y=mean(y)) %>% filter(Species =="obesus") %>% 
  filter(`Tributary/Collection Location`!="Flowing Well #7")

```

```{r}
n<-g + 
  geom_tippoint(aes(fill=Taxon, shape=Taxon), cex=5, alpha=1.0) +
  geom_text_repel(data=olabels, aes(x=x, y=y, label=`Tributary/Collection Location`, max.overalps=Inf)) +
  geom_text_repel(data=labels, aes(x=x, y=y, label= Species)) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  scale_shape_manual(values=c(22,22,23,25,25,24,24,21,21,21))


n
ggsave("outputs/107/438-network-05.pdf", width=10, height=6)

```
# Alternative maf of 0.10


```{r}
net<-read.nexus.networx("outputs/107/438-10.network")
```


```{r}
g<-ggsplitnet(net, color="grey50")  
g$data<-left_join(g$data, nm, by=c("label"="Sample")) 

labels<-g$data %>% filter(isTip==TRUE) %>% group_by(Species) %>% summarize(x=mean(x), y=mean(y)) %>% filter(Species !="obesus")
labels2<-g$data %>% filter(isTip==TRUE) %>% group_by(Species, `Tributary/Collection Location`) %>% summarize(x=mean(x), y=mean(y)) %>% filter(Species !="obesus")

olabels<-g$data %>% filter(isTip==TRUE) %>% group_by(Vicinity, `Tributary/Collection Location`, Species) %>% summarize(x=mean(x), y=mean(y)) %>% filter(Species =="obesus") %>% 
  filter(`Tributary/Collection Location`!="Flowing Well #7")

```

```{r}
n<-g + 
  geom_tippoint(aes(fill=Taxon, shape=Taxon), cex=5, alpha=1.0) +
  geom_text_repel(data=olabels, aes(x=x, y=y, label=`Tributary/Collection Location`, max.overalps=Inf)) +
  geom_text_repel(data=labels, aes(x=x, y=y, label= Species)) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  scale_shape_manual(values=c(22,22,23,25,25,24,24,21,21,21))


n

ggsave("outputs/107/438-network-10.pdf", width=10, height=6)

```


#Now to properly bootstrap an svdquartets tree with more sampling of quartets. 

Choosing... maf 10 not varsites.
conda create -y -n py27 python=2.7 
conda activate py27
conda install python2.7-dev
srun -p high --nodes=1 --ntasks=1 --cpus-per-task=6 --time=36:00:00 paup4a168_ubuntu64 svd-10-boot1000.nex 

problem install dev libraris and python, running 1000 bootstraps in the background 



```{r}
#s<-read.nexus("outputs/107/svdq-boot-05-nomin-boot1000.tre")
s<-read.nexus("outputs/107/svdq-05-nomin-boot1000.trees")
s<-consensus(s, p=0.5) # p can't be less than .5

s<-root(s, c("P_macrolepidotus"))

#ggtree(s, branch.length = 'none') %<+% block +
 # geom_tippoint(aes(shape=Species, fill=Subspecies, size=Count)) + 

tt<-ggtree(s)
tu<-tt$data %>% filter(isTip==FALSE) %>% filter(label != "1")
tu$label<-as.numeric(tu$label)

ggtree(s, branch.length = "none") %<+% st +  
  geom_text(data=tu, aes(x=x, y=y, label=round(label,2)*100)) +
  geom_tippoint(aes(fill=Taxon, shape=Taxon), cex=4) + 
  geom_tiplab(size=4, offset = 0.5) +
  xlim(0,20) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  scale_shape_manual(values=c(22,22,23,25,25,24,24,21,21,21)) 



ggsave("outputs/107/svdq-maf05-boot1000.pdf", width=10, height = 8.5)
ggsave("outputs/107/svdq-maf05-boot1000.jpeg", width=10, height = 8.5)

```


## Access meta and filter

```{r}
mall<-read_csv("meta/05012024.csv")
colls<-read_csv("meta/tui-chub-georeferenced-edited.csv")
counts<-mall %>% filter(Vicinity %in% phylo$Vicinity) %>% group_by(Vicinity, `Tributary/Collection Location`) %>% summarize(Count=n()) %>% left_join(colls, by=c(`Tributary/Collection Location`="Name"))
counts

view(counts)
```



