---
title: "108-recalled-snps"
output: html_document
date: "2024-05-06"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(ggtree)
```


Had good results with the 532 data set for treemix. Will recall, after some filtering for sites in obesus, including Pogonichthys and Cowhead Lake fish.

Note to self to double check pogonichthys.

So, 543 inds +  G. orcuttii P. macrolepidotus, then filtering for small sample sizes.

```{r}
calls<-read_csv("meta/phylo-582.csv") %>% 
  filter(`Tributary/Collection Location` != "3/8 Mile Spring")
```

582 inds w/o "3/8 Mile Spring".   

```{r}
clst<-calls %>% mutate(Inter=GVL_Code) %>% mutate(Cluster=paste0(Species,"_",`Tributary/Collection Location`)) 
clst$Cluster<-gsub(" ","_",clst$Cluster)

clst<-clst %>% filter(!Cluster %in% c("obesus_Little_Hot_Creek_Pond","obesus_Little_Hot_Creek","obesus_Upper_Owens_Gorge","obesus_SE_Pond_(White_Mountain_Research_Center",
                                      "obesus_SW_Pond_(White_Mountain_Research_Center")) %>% group_by(Cluster) %>% mutate(Count=n())
clst1<-filter(clst, Count > 2)
clst<-clst %>% filter(Cluster=="G._orcuttii_Desert_Discovery_Pond") %>% bind_rows(clst1)
clst %>% summarize(Count=n())

clst2<-clst %>% mutate(Intra="1")
clst2$Sample<-gsub("_1","",clst2$Sample)
clst2
```

```{r}
clst2 %>% dplyr::ungroup() %>% dplyr::select(Sample) %>% write_tsv(file="meta/544.names", col_names=FALSE)
clst2 %>% dplyr::ungroup() %>% dplyr::select(Path2) %>% write_tsv(file="meta/544.bamlist", col_names=FALSE)
```

95% thresh, 544*.95
```{sh, eval=FALSE}
srun -t 36:00:00 -p bigmemh --mem=128G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 24 \
-minInd 517 -bam bamlists/544.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-out outputs/108/snps-544-01  \
-minMaf 0.01 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/108/snps-544-01.out 2> outputs/108/snps-544-01.err &
```

