---
title: "203-owens-planning"
output: html_document
date: "2025-02-13"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(pcadapt)
```

```{r}
obs5<-read_csv("meta/obesus-500k-meta.csv") %>% mutate(Path2=ifelse(Dedup > 2e6, paste0("data/downsample/",Ind,".reduced.bam"),
                     Path)) %>% mutate(Taxon=Species) %>% filter(Species %in% c("obesus","snyderi","toikona"))
                     
```

```{r, echo=FALSE}
PlotAdmix<- function(file, meta) {

q<-read_delim(file, delim=" ", col_names=FALSE)
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Last col empty
q<-q[1:length(colnames(q))-1]

#Bind met and arrange 
df<-bind_cols(q, meta) %>% mutate(Region=Vicinity) %>% arrange(Species, Region) %>% mutate(Index=1:n())
df$Region<-factor(df$Region, levels=unique(df$Region))

rdf<-df %>% dplyr::select(Species, Region, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 4:(4+length(colnames(q))-1))

#Make names for structure-like plot
labels<-rdf %>% group_by(Species,Region) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Region,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  ylab("Q\n") +
  theme(legend.position = "") +
  theme(axis.text = element_text(size=10, face="bold")) +
  theme(axis.title.y=element_text(size=14, face="bold")) +
  scale_x_continuous(breaks=labels$Position, labels=labels$Region) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  #theme(axis.text.x=element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  theme(axis.ticks.x = element_blank()) +
  facet_grid(.~Species, scales = "free_x", space="free")

}
```

```{r}
PlotAdmix("outputs/104/obs5-K2.qopt", obs5)
PlotAdmix("outputs/104/obs5-K3.qopt", obs5)
PlotAdmix("outputs/104/obs5-K4.qopt", obs5)
```


```{r}
PlotAdmix("outputs/104/obs5-K5.qopt", obs5)
```

## Visualizing individual admixture across years

```{r, eval=TRUE}
file<-"~/github/fish-lake/outputs/104/obs5-K5.qopt"

q<-read_delim(file, delim=" ", col_names=FALSE)   

meta<-obs5
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Last col empty
q<-q[1:length(colnames(q))-1]

#Bind met and arrange 
df<-bind_cols(q, meta) %>% mutate(Region=Vicinity) %>% filter(Region %in% c("Mammoth Mountain","Owens River")) %>% filter(Dedup > 5e5) %>% arrange(Region, `Tributary/Collection Location`, Year) %>% mutate(Index=1:n())
df$Region<-factor(df$Region, levels=unique(df$Region))
df %>% write_csv("meta/owens-samples-500K.csv")
rdf<-df %>% dplyr::select(Region, `Tributary/Collection Location`, Index, Year, colnames(q) ) %>% gather(key=Ancestry, value=Q, 5:(5+length(colnames(q))-1)) 

#Make names for structure-like plot
labels<-rdf %>% group_by(`Tributary/Collection Location`, Year) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(`Tributary/Collection Location`, Year, Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  ylab("Q\n") +
 # theme(legend.position = "") +
  theme(axis.text = element_text(size=10, face="bold")) +
  theme(axis.title.y=element_text(size=14, face="bold")) +
  scale_x_continuous(breaks=labels$Position, labels=paste0(labels$`Tributary/Collection Location`," ",labels$Year)) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  #theme(axis.text.x=element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  theme(axis.ticks.x = element_blank()) 

ggsave("outputs/203/owens-admix-k5.pdf", width=12, height=8)
```


## Redo


Downsample to fewer individuals/pop and recreate the proposed genomics work?

Try admixture program?

Try to remove closely related individuals?

```{r}
trial<-meta %>% mutate(Region=Vicinity) %>% filter(Region %in% c("Mammoth Mountain","Owens River")) %>% filter(Dedup > 5e5) %>% arrange(Region, `Tributary/Collection Location`, Year) %>% mutate(Index=1:n())

ogs<-meta %>% filter(Vicinity %in% c("Lassen County","Walker Lake" )) %>% filter(Dedup>5e5)

test<-bind_rows(trial,ogs) 
test %>% select(Path2) %>% write_tsv(col_names = FALSE, file="bamlists/test174.bamlist")
test %>% select(GVL_Code) %>% write_tsv(col_names = FALSE, file="bamlists/names174.txt")

```

Call snps and create GL file

95% of individuals

```{sh, eval=FALSE}
srun -t 24:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 165 -bam bamlists/test174.bamlist -ref /home/maccamp/genomes/gila-orcutti/GCA_026230005.1_fGilOrc1.0.hap1_genomic.fna.gz \
-out outputs/203/snps  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doGlf 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/203/snps.out 2> outputs/203/snps.err &

plink --tped snps.tped --tfam snps.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz

#32811 variants and 174 people pass filters and QC.

bcftools reheader --samples bamlists/names174.txt -o outputs/203/renamed.vcf.gz outputs/203/plink.vcf.gz

#Using random seed: 1739489321

bcftools +fill-tags outputs/203/renamed.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/203/pruned-maf05-45.vcf

#Using random seed: 1739490854

bcftools +fill-tags outputs/203/renamed.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.10' | bcftools view -i 'INFO/MAF < 0.45' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/203/pruned-maf10-45.vcf

#for admixture

convert to plink, and convert some more
plink --vcf pruned-maf05-45.vcf --make-bed --out myplink --allow-extra-chr --noweb
plink -bfile myplink --recode12 -out myplink2 --allow-extra-chr --noweb

for K in 1 2 3 4 5 6 7 8; \
do ~/github/admixture/admixture --cv myplink2.ped $K | tee log${K}.out; done
#scores
grep -h CV log*.out | cut -f 3,4 -d ' '  | perl -pe 's/://g' | perl -pe s'/\(|\)//g' > scores.txt 

# in subdir maf10
convert to plink, and convert some more
plink --vcf pruned-maf10-45.vcf --make-bed --out myplink --allow-extra-chr --noweb
plink -bfile myplink --recode12 -out myplink2 --allow-extra-chr --noweb


```

## PCA


```{r}
path_to_file <- "outputs/203/pruned-maf05-45.vcf"
path_to_file <- "outputs/203/maf10/pruned-maf10-45.vcf"

filename <- read.pcadapt(path_to_file, type = "vcf")
x <- pcadapt(input = filename, K = 20) 
```

```{r}
panela<-plot(x, option = "screeplot", K = 10) + ggtitle("A") +
  theme(plot.title = element_text(size=14, face="bold")) +
  theme(panel.grid = element_blank(), panel.background = element_blank()) +
  scale_x_continuous(breaks=seq(1,10,1), labels=seq(1,10,1)) +
  ylab("Proption of Explained Variance\n") +
  xlab("\nPrincipal Component") +
  theme(axis.title=element_text(size=12))
panela

```
```{r}
plot(x, option = "scores", pop = factor(test$`Tributary/Collection Location`)) 
```

```{r}
scores<-read_delim("outputs/203/maf10/scores.txt", col_names = c("K","Cross-Validation Error")) %>% mutate(x=1:n())

ggplot(scores) +
  geom_line(aes(x=x, y=`Cross-Validation Error`)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks=(1:nrow(scores)))
```

## Admix

```{r, echo=FALSE}
PlotAdmixture<- function(file, meta) {

q<-read_delim(file, delim=" ", col_names=FALSE)
#Make generic colnames

nums<-1:length(colnames(q))
mynames<-paste0("Q",nums)

qs<-length(colnames(q))-1

colnames(q)<-mynames

#Bind met and arrange 
df<-bind_cols(q, meta) %>% arrange(Species, `Tributary/Collection Location`, Year) %>% mutate(Index=1:n())
df$`Tributary/Collection Location`<-factor(df$`Tributary/Collection Location`, levels=unique(df$`Tributary/Collection Location`))

rdf<-df %>% dplyr::select(Species, `Tributary/Collection Location`, Year, Index, colnames(q) ) %>% gather(key=Ancestry, value=Q, 5:(5+length(colnames(q))-1))

#Make names for structure-like plot
labels<-rdf %>% group_by(`Tributary/Collection Location`, Year) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(`Tributary/Collection Location`,Year, Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

#Plot
ggplot(rdf) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=labels, x = labels$Start - 0.5, y=0, xend = labels$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=labels, x = labels$Stop[length(labels$Stop)]  + 0.5, y=0, 
               xend= labels$Stop[length(labels$Stop)] + 0.5, yend=1,  alpha=0.9,
               size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0+0.5, xend= labels$Stop[length(labels$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, labels$Stop[length(labels$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") +
  ylab("Q\n") +
  theme(legend.position = "") +
  theme(axis.text = element_text(size=10, face="bold")) +
  theme(axis.title.y=element_text(size=14, face="bold")) +
  scale_x_continuous(breaks=labels$Position, labels=paste0(labels$`Tributary/Collection Location`," ",labels$Year)) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  #theme(axis.text.x=element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  theme(axis.ticks.x = element_blank())

}
```


```{r}
PlotAdmixture("outputs/203/maf10/myplink2.3.Q", test)
```

```{r}
PlotAdmixture("outputs/203/myplink2.3.Q", test)
```

```{r}
PlotAdmixture("outputs/203/myplink2.5.Q", test)
```


